<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-09-18" />

<title>Checking how CpG methylation affects TF binding</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiberseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/fiberseq">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Checking how CpG methylation affects TF
binding</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-09-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>fiberseq/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250831code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250831)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250831code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250831)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229fiberseqtreeefbd42b4eff4fe80d85379fe3b1b5a972a2de52ftargetblankefbd42ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/fiberseq/tree/efbd42b4eff4fe80d85379fe3b1b5a972a2de52f" target="_blank">efbd42b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229fiberseqtreeefbd42b4eff4fe80d85379fe3b1b5a972a2de52ftargetblankefbd42ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/fiberseq/tree/efbd42b4eff4fe80d85379fe3b1b5a972a2de52f" target="_blank">efbd42b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/TF_co-acc_NRF1_cpg2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/TF_co-acc_NRF1_cpg2.Rmd</code>)
and HTML (<code>docs/TF_co-acc_NRF1_cpg2.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/efbd42b4eff4fe80d85379fe3b1b5a972a2de52f/analysis/TF_co-acc_NRF1_cpg2.Rmd" target="_blank">efbd42b</a>
</td>
<td>
XSun
</td>
<td>
2025-09-18
</td>
<td>
updae
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(knitr)
library(data.table)
library(reshape2)
library(GenomicRanges)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(Biostrings)
library(gridExtra)
library(purrr)

source(&quot;/project/spott/kevinluo/Fiber_seq/fiberhub/code/process_fiberseq_data.R&quot;)
source(&quot;/project/spott/kevinluo/Fiber_seq/fiberhub/code/plots.R&quot;)

DT::datatable(matrix())</code></pre>
<div id="htmlwidget-55d484e2ce688c3024c2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-55d484e2ce688c3024c2">{"x":{"filter":"none","vertical":false,"data":[[null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We check how CpG methylation affects TF binding here.</p>
<p>We focus on the regions that only contains one NRF1 motif site.</p>
<p>Details for NRF1 motif: <a
href="https://jaspar.elixir.no/matrix/MA0506.1/"
class="uri">https://jaspar.elixir.no/matrix/MA0506.1/</a></p>
<p>We just take the consensus into consideration, there are 2 CpG sites
that may be methylated: position 2 and position 8. On the opposite
strand, they are position 3 and position 9.</p>
<figure class="half">
<img src="https://github.com/xsun1229/fiberseq/raw/master/output/MA0506.1.png" width="100%">
</figure>
</div>
<div id="examples" class="section level1">
<h1>Examples</h1>
<p>We illustrate the workflow using several examples.</p>
<pre class="r"><code>TF1 &lt;- &quot;NRF1&quot;

# ---- Load motif sites ----
motif_sites &lt;- readRDS(file.path(
  &quot;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs&quot;,
  paste0(TF1, &quot;.GM12878.sites.chip.labels.rds&quot;))
) %&gt;% filter(chip_label == 1, !is.na(start))

gr_motif &lt;- GRanges(
  seqnames = motif_sites$chr,
  ranges   = IRanges(start = motif_sites$start + 100,
                     end   = motif_sites$end   - 100),
  site     = motif_sites$name
)

# ---- Find motif pairs ----
dist_cutoff_large &lt;- 300L
dist_cutoff_small &lt;- 0L

hits &lt;- findOverlaps(gr_motif, gr_motif, ignore.strand = TRUE, maxgap = dist_cutoff_large)
q &lt;- queryHits(hits); s &lt;- subjectHits(hits)

keep &lt;- q &lt; s
q &lt;- q[keep]; s &lt;- s[keep]
d &lt;- distance(gr_motif[q], gr_motif[s])

sel &lt;- d &gt;= dist_cutoff_small &amp; d &lt;= dist_cutoff_large
q &lt;- q[sel]; s &lt;- s[sel]; d &lt;- d[sel]

df_close &lt;- data.frame(
  motif1_site  = gr_motif[q]$site,
  motif2_site  = gr_motif[s]$site,
  distance     = as.integer(d)
)

# ---- Motifs involved in &quot;close&quot; pairs ----
motifs_with_neighbors &lt;- unique(c(df_close$motif1_site, df_close$motif2_site))

# ---- Opposite case: motifs without neighbors ----
singleton_motifs &lt;- motif_sites %&gt;%
  filter(!(name %in% motifs_with_neighbors))

df_final &lt;- singleton_motifs %&gt;%
  group_by(chr, name, start, end) %&gt;%
  ungroup() %&gt;%
  transmute(
    chr,
    start = trimws(format(floor(start / 100) * 100 - 500, scientific = FALSE)),
    end   = trimws(format(ceiling(end / 100) * 100 + 500, scientific = FALSE))
  ) %&gt;%
  mutate(chr_num = as.numeric(gsub(&quot;chr&quot;, &quot;&quot;, chr))) %&gt;%
  arrange(chr_num, as.numeric(start)) %&gt;%
  select(-chr_num)



df_final$id &lt;- paste0(df_final$chr,&quot;-&quot;, df_final$start, &quot;-&quot;,df_final$end)
df_final &lt;- df_final[!duplicated(df_final$id),]

out_dir=&quot;/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_singleton/&quot;
sample_name=&quot;LCL_18508&quot;

rid_motifsite_mat           &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/rid_motifsite_mat_sites_select.RDS&quot;)
rid_motifsite_combined      &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/rid_motifsite_combined_sites_select.RDS&quot;)
rid_motifsite_methy         &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/rid_motifsite_methy_sites_select.RDS&quot;)
sites_selected              &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/sites_select.RDS&quot;)
RIDs_access.df_all          &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/RIDs_access.df_sites_select.RDS&quot;)
cpg_status_all              &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/cpg_status_all_sites_select.RDS&quot;)
contingency_table_pos1_all   &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/contingency_table_pos1.RDS&quot;)
contingency_table_pos2_all   &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/contingency_table_pos2.RDS&quot;)
prop_col_pos1_all            &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/contingency_prop_col_pos1.RDS&quot;)
prop_col_pos2_all            &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/contingency_prop_col_pos2.RDS&quot;)</code></pre>
<div id="site10635" class="section level2">
<h2>site10635</h2>
<pre class="r"><code>site &lt;- &quot;site10635&quot;

index_region &lt;- which(singleton_motifs$name == site)

test_range_str &lt;- paste0(df_final$chr[index_region],&quot;:&quot;,df_final$start[index_region],&quot;-&quot;,df_final$end[index_region])
plot_name &lt;- paste0(TF1,&quot;_pairs_&quot;,sample_name,&quot;_&quot;,test_range_str)
result_dir &lt;- paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)

region &lt;- get_region(result_dir)
subtitle &lt;- sprintf(&quot;%s:%s-%s&quot;, region$chr, region$start, region$end)

# motif coordinates
start &lt;- singleton_motifs$start[index_region] + 100
end   &lt;- singleton_motifs$end[index_region] -100

region_plot &lt;- list(chr = df_final$chr[index_region],
                    start = start - 150,
                    end = end + 150)

res &lt;- try(get_region_result(result_dir,
                             filter_fire_reads = FALSE,
                             cover_region_only = FALSE,
                             select_A_only = FALSE,
                             split_haplotype = FALSE,
                             fiberHMM_size_breaks = c(60,100),
                             fiberHMM_cols = c(&quot;magenta&quot;,&quot;royalblue&quot;,&quot;darkgreen&quot;),
                             highlight_window = data.frame(start=start, end=end),
                             color_highlight_window = &quot;#44d6d8ff&quot;,
                             plot_name = plot_name,
                             # cluster = &quot;fiberHMM_configs&quot;,
                             # cluster_regions = cluster_regions,
                             show_RIDs = F,
                             region = region_plot,
                             subtitle = subtitle,
                             verbose = T)
)</code></pre>
<p>Get region result… Loading FiberHMM result from
/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_singleton//NRF1_pairs_LCL_18508_chr6:36546800-36548100/parsed
fiberHMM size levels: &lt; 60 bp 60-100 bp &gt; 100 bp Clustering reads…
34 reads. highlight windows: start end 1 36547473 36547484</p>
<pre class="r"><code>gg &lt;- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = &quot;v&quot;,
  axis = &quot;b&quot;
))</code></pre>
<p><img src="figure/TF_co-acc_NRF1_cpg2.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;We identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. \n&quot;)</code></pre>
<p>We identify CpG positions within this motif. The CpG positions are
extracted from the corresponding .fa files.</p>
<pre class="r"><code>cat(&quot;We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome&quot;)</code></pre>
<p>We restrict the analysis to reads without nucleosome footprints,
since transcription factors cannot bind when the motif is occupied by a
nucleosome</p>
<pre class="r"><code>DT::datatable(rid_motifsite_mat[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;CpG positions in the motif for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0c375f72d786c94bb27a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c375f72d786c94bb27a">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">CpG positions in the motif for each read --site10635<\/caption>","data":[["m64403e_220916_085351/117376538/ccs","m64403e_220916_085351/131007477/ccs","m64403e_220916_085351/70322097/ccs","m64403e_220916_085351/94701919/ccs","m64403e_230429_051256/105645905/ccs","m64403e_230429_051256/112133133/ccs","m64403e_230429_051256/118884515/ccs","m64403e_230429_051256/130550270/ccs","m64403e_230429_051256/161416462/ccs","m64403e_230429_051256/163578214/ccs","m64403e_230429_051256/31393476/ccs","m64403e_230429_051256/81331886/ccs","m64403e_230429_051256/83626748/ccs","m64403e_230430_142929/116720980/ccs","m64403e_230430_142929/14483779/ccs","m64403e_230430_142929/145228528/ccs","m64403e_230430_142929/168953860/ccs","m64403e_230430_142929/25954792/ccs","m64403e_230430_142929/62457528/ccs","m64403e_230430_142929/64422004/ccs","m64403e_230430_142929/78382458/ccs","m64403e_230430_142929/80021862/ccs","m64403e_230430_142929/87426760/ccs","m64403e_230501_235831/102303843/ccs","m64403e_230501_235831/112001910/ccs","m64403e_230501_235831/146671840/ccs","m64403e_230501_235831/155648990/ccs","m64403e_230501_235831/159515030/ccs","m64403e_230501_235831/159973437/ccs","m64403e_230501_235831/169937137/ccs","m64403e_230501_235831/174850765/ccs","m64403e_230501_235831/59245122/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>36547473<\/th>\n      <th>36547474<\/th>\n      <th>36547475<\/th>\n      <th>36547476<\/th>\n      <th>36547477<\/th>\n      <th>36547478<\/th>\n      <th>36547479<\/th>\n      <th>36547480<\/th>\n      <th>36547481<\/th>\n      <th>36547482<\/th>\n      <th>36547483<\/th>\n      <th>36547484<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;We then retrieve methylation information from res$read. \n&quot;)</code></pre>
<p>We then retrieve methylation information from res$read.</p>
<pre class="r"><code>DT::datatable(rid_motifsite_methy[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation quality score for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-9fa6da26214ebfd05e61" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9fa6da26214ebfd05e61">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation quality score for each read --site10635<\/caption>","data":[["m64403e_220916_085351/117376538/ccs","m64403e_220916_085351/131007477/ccs","m64403e_220916_085351/70322097/ccs","m64403e_220916_085351/94701919/ccs","m64403e_230429_051256/105645905/ccs","m64403e_230429_051256/112133133/ccs","m64403e_230429_051256/118884515/ccs","m64403e_230429_051256/130550270/ccs","m64403e_230429_051256/161416462/ccs","m64403e_230429_051256/163578214/ccs","m64403e_230429_051256/31393476/ccs","m64403e_230429_051256/81331886/ccs","m64403e_230429_051256/83626748/ccs","m64403e_230430_142929/116720980/ccs","m64403e_230430_142929/14483779/ccs","m64403e_230430_142929/145228528/ccs","m64403e_230430_142929/168953860/ccs","m64403e_230430_142929/25954792/ccs","m64403e_230430_142929/62457528/ccs","m64403e_230430_142929/64422004/ccs","m64403e_230430_142929/78382458/ccs","m64403e_230430_142929/80021862/ccs","m64403e_230430_142929/87426760/ccs","m64403e_230501_235831/102303843/ccs","m64403e_230501_235831/112001910/ccs","m64403e_230501_235831/146671840/ccs","m64403e_230501_235831/155648990/ccs","m64403e_230501_235831/159515030/ccs","m64403e_230501_235831/159973437/ccs","m64403e_230501_235831/169937137/ccs","m64403e_230501_235831/174850765/ccs","m64403e_230501_235831/59245122/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,0.7558594,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.7792969,null,null,0.8808594,null,null,null,0.8339844,null,null,null,null],[null,null,null,null,0.8808594,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[0.7128906,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.7207031,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.7441406,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>36547473<\/th>\n      <th>36547474<\/th>\n      <th>36547475<\/th>\n      <th>36547476<\/th>\n      <th>36547477<\/th>\n      <th>36547478<\/th>\n      <th>36547479<\/th>\n      <th>36547480<\/th>\n      <th>36547481<\/th>\n      <th>36547482<\/th>\n      <th>36547483<\/th>\n      <th>36547484<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mC.\n&quot;)</code></pre>
<p>By combining the CpG position matrix with the methylation quality
matrix, we annotate methylated sites as mC.</p>
<pre class="r"><code>DT::datatable(rid_motifsite_combined[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation annotation for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-1b1a8ec0941961691c13" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1b1a8ec0941961691c13">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation annotation for each read --site10635<\/caption>","data":[["m64403e_220916_085351/117376538/ccs","m64403e_220916_085351/131007477/ccs","m64403e_220916_085351/70322097/ccs","m64403e_220916_085351/94701919/ccs","m64403e_230429_051256/105645905/ccs","m64403e_230429_051256/112133133/ccs","m64403e_230429_051256/118884515/ccs","m64403e_230429_051256/130550270/ccs","m64403e_230429_051256/161416462/ccs","m64403e_230429_051256/163578214/ccs","m64403e_230429_051256/31393476/ccs","m64403e_230429_051256/81331886/ccs","m64403e_230429_051256/83626748/ccs","m64403e_230430_142929/116720980/ccs","m64403e_230430_142929/14483779/ccs","m64403e_230430_142929/145228528/ccs","m64403e_230430_142929/168953860/ccs","m64403e_230430_142929/25954792/ccs","m64403e_230430_142929/62457528/ccs","m64403e_230430_142929/64422004/ccs","m64403e_230430_142929/78382458/ccs","m64403e_230430_142929/80021862/ccs","m64403e_230430_142929/87426760/ccs","m64403e_230501_235831/102303843/ccs","m64403e_230501_235831/112001910/ccs","m64403e_230501_235831/146671840/ccs","m64403e_230501_235831/155648990/ccs","m64403e_230501_235831/159515030/ccs","m64403e_230501_235831/159973437/ccs","m64403e_230501_235831/169937137/ccs","m64403e_230501_235831/174850765/ccs","m64403e_230501_235831/59245122/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["C","C","C","C","C","mC","C","C","C","C","C","C","C","C","C","C","C","C","C","C","mC","C","C","mC","C","C","C","mC","C","C","C","C"],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["mC","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","mC","C","C","C","C","C","C","C","C","C","C"],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>36547473<\/th>\n      <th>36547474<\/th>\n      <th>36547475<\/th>\n      <th>36547476<\/th>\n      <th>36547477<\/th>\n      <th>36547478<\/th>\n      <th>36547479<\/th>\n      <th>36547480<\/th>\n      <th>36547481<\/th>\n      <th>36547482<\/th>\n      <th>36547483<\/th>\n      <th>36547484<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;We summarize the methylation and TF footprint status for each read\n&quot;)</code></pre>
<p>We summarize the methylation and TF footprint status for each
read</p>
<pre class="r"><code>DT::datatable(RIDs_access.df_all[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation and TF footprint status for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-d3bdb3a70b69fe9ab76d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d3bdb3a70b69fe9ab76d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation and TF footprint status for each read --site10635<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["m64403e_220916_085351/117376538/ccs","m64403e_220916_085351/131007477/ccs","m64403e_230429_051256/112133133/ccs","m64403e_230429_051256/118884515/ccs","m64403e_230429_051256/130550270/ccs","m64403e_230429_051256/161416462/ccs","m64403e_230429_051256/163578214/ccs","m64403e_230429_051256/31393476/ccs","m64403e_230429_051256/83626748/ccs","m64403e_230430_142929/62457528/ccs","m64403e_230430_142929/80021862/ccs","m64403e_230501_235831/112001910/ccs","m64403e_230501_235831/155648990/ccs","m64403e_230501_235831/159973437/ccs","m64403e_230501_235831/174850765/ccs","m64403e_230501_235831/59245122/ccs"],["FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","TRUE","TRUE","TRUE","TRUE"],["FALSE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE"],["TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>RID<\/th>\n      <th>TF<\/th>\n      <th>pos1<\/th>\n      <th>pos2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;For position 1, we have\n&quot;)</code></pre>
<p>For position 1, we have</p>
<pre class="r"><code>kable(as.matrix(contingency_table_pos1_all[[site]]),
      caption = paste(&quot;Contingency table for&quot;, site))</code></pre>
<table>
<caption>Contingency table for site10635</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">13</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;The proportion of TF footprint in different methylation groups&quot;)</code></pre>
<p>The proportion of TF footprint in different methylation groups</p>
<pre class="r"><code>kable(as.matrix(prop_col_pos1_all[[site]]),
      caption = paste(&quot;Proportion table for&quot;, site))</code></pre>
<table>
<caption>Proportion table for site10635</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">13.33</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">86.67</td>
<td align="right">100</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;For position 2, we have\n&quot;)</code></pre>
<p>For position 2, we have</p>
<pre class="r"><code>kable(as.matrix(contingency_table_pos2_all[[site]]),
      caption = paste(&quot;Contingency table for&quot;, site))</code></pre>
<table>
<caption>Contingency table for site10635</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">13</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;The proportion of TF footprint in different methylation groups&quot;)</code></pre>
<p>The proportion of TF footprint in different methylation groups</p>
<pre class="r"><code>kable(as.matrix(prop_col_pos2_all[[site]]),
      caption = paste(&quot;Proportion table for&quot;, site))</code></pre>
<table>
<caption>Proportion table for site10635</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos2_FALSE</th>
<th align="right">CpG_pos2_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">7.14</td>
<td align="right">50</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">92.86</td>
<td align="right">50</td>
</tr>
</tbody>
</table>
</div>
<div id="site11369" class="section level2">
<h2>site11369</h2>
<pre class="r"><code>site &lt;- &quot;site11369&quot;

index_region &lt;- which(singleton_motifs$name == site)

test_range_str &lt;- paste0(df_final$chr[index_region],&quot;:&quot;,df_final$start[index_region],&quot;-&quot;,df_final$end[index_region])
plot_name &lt;- paste0(TF1,&quot;_pairs_&quot;,sample_name,&quot;_&quot;,test_range_str)
result_dir &lt;- paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)

region &lt;- get_region(result_dir)
subtitle &lt;- sprintf(&quot;%s:%s-%s&quot;, region$chr, region$start, region$end)

# motif coordinates
start &lt;- singleton_motifs$start[index_region] + 100
end   &lt;- singleton_motifs$end[index_region] -100

region_plot &lt;- list(chr = df_final$chr[index_region],
                    start = start - 150,
                    end = end + 150)

res &lt;- try(get_region_result(result_dir,
                             filter_fire_reads = FALSE,
                             cover_region_only = FALSE,
                             select_A_only = FALSE,
                             split_haplotype = FALSE,
                             fiberHMM_size_breaks = c(60,100),
                             fiberHMM_cols = c(&quot;magenta&quot;,&quot;royalblue&quot;,&quot;darkgreen&quot;),
                             highlight_window = data.frame(start=start, end=end),
                             color_highlight_window = &quot;#44d6d8ff&quot;,
                             plot_name = plot_name,
                             # cluster = &quot;fiberHMM_configs&quot;,
                             # cluster_regions = cluster_regions,
                             show_RIDs = F,
                             region = region_plot,
                             subtitle = subtitle,
                             verbose = T)
)</code></pre>
<p>Get region result… Loading FiberHMM result from
/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_singleton//NRF1_pairs_LCL_18508_chr6:151451500-151452800/parsed
fiberHMM size levels: &lt; 60 bp 60-100 bp &gt; 100 bp Clustering reads…
34 reads. highlight windows: start end 1 151452155 151452166</p>
<pre class="r"><code>gg &lt;- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = &quot;v&quot;,
  axis = &quot;b&quot;
))</code></pre>
<p><img src="figure/TF_co-acc_NRF1_cpg2.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;We identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. \n&quot;)</code></pre>
<p>We identify CpG positions within this motif. The CpG positions are
extracted from the corresponding .fa files.</p>
<pre class="r"><code>cat(&quot;We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome&quot;)</code></pre>
<p>We restrict the analysis to reads without nucleosome footprints,
since transcription factors cannot bind when the motif is occupied by a
nucleosome</p>
<pre class="r"><code>DT::datatable(rid_motifsite_mat[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;CpG positions in the motif for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-e41253b73217ca07c332" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e41253b73217ca07c332">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">CpG positions in the motif for each read --site11369<\/caption>","data":[["m64403e_220914_215953/9241202/ccs","m64403e_220916_085351/105840831/ccs","m64403e_220916_085351/139200152/ccs","m64403e_220916_085351/97583211/ccs","m64403e_230429_051256/112199023/ccs","m64403e_230429_051256/134743343/ccs","m64403e_230429_051256/143328066/ccs","m64403e_230429_051256/165021248/ccs","m64403e_230429_051256/178258603/ccs","m64403e_230429_051256/43977015/ccs","m64403e_230429_051256/5507436/ccs","m64403e_230429_051256/57346017/ccs","m64403e_230429_051256/67569529/ccs","m64403e_230429_051256/67896843/ccs","m64403e_230429_051256/85459329/ccs","m64403e_230429_051256/95224039/ccs","m64403e_230429_051256/95815949/ccs","m64403e_230430_142929/115344690/ccs","m64403e_230430_142929/117835257/ccs","m64403e_230430_142929/127534276/ccs","m64403e_230430_142929/134807582/ccs","m64403e_230430_142929/152109369/ccs","m64403e_230430_142929/174065518/ccs","m64403e_230430_142929/27788940/ccs","m64403e_230430_142929/2951042/ccs","m64403e_230430_142929/45352319/ccs","m64403e_230430_142929/52495570/ccs","m64403e_230430_142929/64815875/ccs","m64403e_230430_142929/85854693/ccs","m64403e_230501_235831/132187311/ccs","m64403e_230501_235831/54789329/ccs","m64403e_230501_235831/56557721/ccs","m64403e_230501_235831/63768095/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>151452155<\/th>\n      <th>151452156<\/th>\n      <th>151452157<\/th>\n      <th>151452158<\/th>\n      <th>151452159<\/th>\n      <th>151452160<\/th>\n      <th>151452161<\/th>\n      <th>151452162<\/th>\n      <th>151452163<\/th>\n      <th>151452164<\/th>\n      <th>151452165<\/th>\n      <th>151452166<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;We then retrieve methylation information from res$read. \n&quot;)</code></pre>
<p>We then retrieve methylation information from res$read.</p>
<pre class="r"><code>DT::datatable(rid_motifsite_methy[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation quality score for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-6b35d4872c33fe5849ab" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6b35d4872c33fe5849ab">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation quality score for each read --site11369<\/caption>","data":[["m64403e_220914_215953/9241202/ccs","m64403e_220916_085351/105840831/ccs","m64403e_220916_085351/139200152/ccs","m64403e_220916_085351/97583211/ccs","m64403e_230429_051256/112199023/ccs","m64403e_230429_051256/134743343/ccs","m64403e_230429_051256/143328066/ccs","m64403e_230429_051256/165021248/ccs","m64403e_230429_051256/178258603/ccs","m64403e_230429_051256/43977015/ccs","m64403e_230429_051256/5507436/ccs","m64403e_230429_051256/57346017/ccs","m64403e_230429_051256/67569529/ccs","m64403e_230429_051256/67896843/ccs","m64403e_230429_051256/85459329/ccs","m64403e_230429_051256/95224039/ccs","m64403e_230429_051256/95815949/ccs","m64403e_230430_142929/115344690/ccs","m64403e_230430_142929/117835257/ccs","m64403e_230430_142929/127534276/ccs","m64403e_230430_142929/134807582/ccs","m64403e_230430_142929/152109369/ccs","m64403e_230430_142929/174065518/ccs","m64403e_230430_142929/27788940/ccs","m64403e_230430_142929/2951042/ccs","m64403e_230430_142929/45352319/ccs","m64403e_230430_142929/52495570/ccs","m64403e_230430_142929/64815875/ccs","m64403e_230430_142929/85854693/ccs","m64403e_230501_235831/132187311/ccs","m64403e_230501_235831/54789329/ccs","m64403e_230501_235831/56557721/ccs","m64403e_230501_235831/63768095/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,0.7910156,null,0.6816406,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,0.7832031,null,null,0.7753906,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[0.8535156,null,null,null,null,null,null,null,null,null,null,null,0.6816406,null,0.7167969,null,null,null,0.8261719,null,null,null,0.7089844,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>151452155<\/th>\n      <th>151452156<\/th>\n      <th>151452157<\/th>\n      <th>151452158<\/th>\n      <th>151452159<\/th>\n      <th>151452160<\/th>\n      <th>151452161<\/th>\n      <th>151452162<\/th>\n      <th>151452163<\/th>\n      <th>151452164<\/th>\n      <th>151452165<\/th>\n      <th>151452166<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mC.\n&quot;)</code></pre>
<p>By combining the CpG position matrix with the methylation quality
matrix, we annotate methylated sites as mC.</p>
<pre class="r"><code>DT::datatable(rid_motifsite_combined[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation annotation for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-6e49b5d231d9a4691769" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6e49b5d231d9a4691769">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation annotation for each read --site11369<\/caption>","data":[["m64403e_220914_215953/9241202/ccs","m64403e_220916_085351/105840831/ccs","m64403e_220916_085351/139200152/ccs","m64403e_220916_085351/97583211/ccs","m64403e_230429_051256/112199023/ccs","m64403e_230429_051256/134743343/ccs","m64403e_230429_051256/143328066/ccs","m64403e_230429_051256/165021248/ccs","m64403e_230429_051256/178258603/ccs","m64403e_230429_051256/43977015/ccs","m64403e_230429_051256/5507436/ccs","m64403e_230429_051256/57346017/ccs","m64403e_230429_051256/67569529/ccs","m64403e_230429_051256/67896843/ccs","m64403e_230429_051256/85459329/ccs","m64403e_230429_051256/95224039/ccs","m64403e_230429_051256/95815949/ccs","m64403e_230430_142929/115344690/ccs","m64403e_230430_142929/117835257/ccs","m64403e_230430_142929/127534276/ccs","m64403e_230430_142929/134807582/ccs","m64403e_230430_142929/152109369/ccs","m64403e_230430_142929/174065518/ccs","m64403e_230430_142929/27788940/ccs","m64403e_230430_142929/2951042/ccs","m64403e_230430_142929/45352319/ccs","m64403e_230430_142929/52495570/ccs","m64403e_230430_142929/64815875/ccs","m64403e_230430_142929/85854693/ccs","m64403e_230501_235831/132187311/ccs","m64403e_230501_235831/54789329/ccs","m64403e_230501_235831/56557721/ccs","m64403e_230501_235831/63768095/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],["C","C","mC","C","mC","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],["C","mC","C","C","mC","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>151452155<\/th>\n      <th>151452156<\/th>\n      <th>151452157<\/th>\n      <th>151452158<\/th>\n      <th>151452159<\/th>\n      <th>151452160<\/th>\n      <th>151452161<\/th>\n      <th>151452162<\/th>\n      <th>151452163<\/th>\n      <th>151452164<\/th>\n      <th>151452165<\/th>\n      <th>151452166<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;We summarize the methylation and TF footprint status for each read\n&quot;)</code></pre>
<p>We summarize the methylation and TF footprint status for each
read</p>
<pre class="r"><code>DT::datatable(RIDs_access.df_all[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation and TF footprint status for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-3f92d051ff887de577da" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3f92d051ff887de577da">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation and TF footprint status for each read --site11369<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29"],["m64403e_220914_215953/9241202/ccs","m64403e_220916_085351/105840831/ccs","m64403e_220916_085351/139200152/ccs","m64403e_220916_085351/97583211/ccs","m64403e_230429_051256/112199023/ccs","m64403e_230429_051256/134743343/ccs","m64403e_230429_051256/143328066/ccs","m64403e_230429_051256/165021248/ccs","m64403e_230429_051256/43977015/ccs","m64403e_230429_051256/57346017/ccs","m64403e_230429_051256/67569529/ccs","m64403e_230429_051256/67896843/ccs","m64403e_230429_051256/85459329/ccs","m64403e_230429_051256/95224039/ccs","m64403e_230429_051256/95815949/ccs","m64403e_230430_142929/115344690/ccs","m64403e_230430_142929/117835257/ccs","m64403e_230430_142929/127534276/ccs","m64403e_230430_142929/134807582/ccs","m64403e_230430_142929/152109369/ccs","m64403e_230430_142929/174065518/ccs","m64403e_230430_142929/27788940/ccs","m64403e_230430_142929/2951042/ccs","m64403e_230430_142929/45352319/ccs","m64403e_230430_142929/64815875/ccs","m64403e_230430_142929/85854693/ccs","m64403e_230501_235831/132187311/ccs","m64403e_230501_235831/54789329/ccs","m64403e_230501_235831/56557721/ccs"],["TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","TRUE","TRUE","TRUE","FALSE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","TRUE","FALSE","TRUE"],["FALSE","FALSE","TRUE","FALSE","TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE"],["FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>RID<\/th>\n      <th>TF<\/th>\n      <th>pos1<\/th>\n      <th>pos2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;For position 1, we have\n&quot;)</code></pre>
<p>For position 1, we have</p>
<pre class="r"><code>kable(as.matrix(contingency_table_pos1_all[[site]]),
      caption = paste(&quot;Contingency table for&quot;, site))</code></pre>
<table>
<caption>Contingency table for site11369</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">5</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">22</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;The proportion of TF footprint in different methylation groups&quot;)</code></pre>
<p>The proportion of TF footprint in different methylation groups</p>
<pre class="r"><code>kable(as.matrix(prop_col_pos1_all[[site]]),
      caption = paste(&quot;Proportion table for&quot;, site))</code></pre>
<table>
<caption>Proportion table for site11369</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">18.52</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">81.48</td>
<td align="right">100</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;For position 2, we have\n&quot;)</code></pre>
<p>For position 2, we have</p>
<pre class="r"><code>kable(as.matrix(contingency_table_pos2_all[[site]]),
      caption = paste(&quot;Contingency table for&quot;, site))</code></pre>
<table>
<caption>Contingency table for site11369</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">5</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">24</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;The proportion of TF footprint in different methylation groups&quot;)</code></pre>
<p>The proportion of TF footprint in different methylation groups</p>
<pre class="r"><code>kable(as.matrix(prop_col_pos2_all[[site]]),
      caption = paste(&quot;Proportion table for&quot;, site))</code></pre>
<table>
<caption>Proportion table for site11369</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos2_FALSE</th>
<th align="right">CpG_pos2_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">17.24</td>
<td align="right">NaN</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">82.76</td>
<td align="right">NaN</td>
</tr>
</tbody>
</table>
<!-- ## chr10:122879000-122880300 -->
<!-- The first example is region chr10:122879500–122879700, focusing on the motif chr10:122879629–122879640. -->
<!-- ```{r message=FALSE,warning=FALSE, fig.height=10, fig.width=10} -->
<!-- library(data.table) -->
<!-- library(reshape2) -->
<!-- library(GenomicRanges) -->
<!-- library(ggplot2) -->
<!-- library(tidyverse) -->
<!-- library(gridExtra) -->
<!-- library(Biostrings) -->
<!-- source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/process_fiberseq_data.R") -->
<!-- source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/plots.R") -->
<!-- test_range_str <- "chr10:122879000-122880300" -->
<!-- plot_name <- "NRF1_pairs_LCL_18508_chr10:122879000-122880300" -->
<!-- result_dir <- "/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_pairs//NRF1_pairs_LCL_18508_chr10:122879000-122880300/parsed" -->
<!-- region <- get_region(result_dir) -->
<!-- subtitle <- sprintf("%s:%s-%s", region$chr, region$start, region$end) -->
<!-- # motif coordinates -->
<!-- start_motif1 <- 122879629 -->
<!-- end_motif1   <- 122879640 -->
<!-- region_plot <- list(chr ="chr10", -->
<!--                     start = 122879500, -->
<!--                     end = 122879800) -->
<!-- res <- try(assemble_region_result(result_dir, -->
<!--                        filter_fire_reads = FALSE, -->
<!--                        cover_region_only = FALSE, -->
<!--                        select_A_only = FALSE, -->
<!--                        split_haplotype = FALSE, -->
<!--                        highlight_window = data.frame(start=c(start_motif1), end=c(end_motif1)), -->
<!--                        color_highlight_window = "#44d6d8ff", -->
<!--                        plot_name = plot_name,region = region_plot, -->
<!--                        subtitle = subtitle) -->
<!--   ) -->
<!-- gg <- res$gg -->
<!-- print(cowplot::plot_grid( -->
<!--   gg$pileup, -->
<!--   gg$dimelo_reads, -->
<!--   gg$fire_fdr, -->
<!--   gg$fire_reads, -->
<!--   gg$fiberHMM, -->
<!--   ncol = 1, -->
<!--   rel_heights = c(4,8,2,8,8), -->
<!--   align = "v", -->
<!--   axis = "b" -->
<!-- )) -->
<!-- ``` -->
<!-- Next, we identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files.  -->
<!-- ```{r} -->
<!-- seq_region <- readDNAStringSet(file.path(result_dir,"/region.fa")) -->
<!-- dna_seq <- seq_region[[1]] -->
<!-- cpg_rel_pos <- start(matchPattern("CG", dna_seq)) -->
<!-- header <- names(seq_region)       # e.g. "chr10:122879000-122880300" -->
<!-- parts <- strsplit(header, "[:-]")[[1]] -->
<!-- chrom <- parts[1] -->
<!-- region_start <- as.integer(parts[2]) -->
<!-- cpg_df <- data.frame( -->
<!--   chrom = chrom, -->
<!--   position = region_start + cpg_rel_pos - 1 -->
<!-- ) -->
<!-- cpg_df_inmotif <- cpg_df[cpg_df$position >= start_motif1 & cpg_df$position <= end_motif1,] -->
<!-- rid_motif1 <- as.character(res$rids_df$RID[res$rids_df$start <= end_motif1 & res$rids_df$end >= start_motif1]) -->
<!-- rid_motifsite_mat <- matrix( -->
<!--   NA,  -->
<!--   nrow = length(rid_motif1),  -->
<!--   ncol = length(start_motif1:end_motif1), -->
<!--   dimnames = list(rid_motif1, as.character(start_motif1:end_motif1)) -->
<!-- ) -->
<!-- rid_motifsite_methy <- rid_motifsite_mat -->
<!-- # Mark CpG positions -->
<!-- rid_motifsite_mat[,colnames(rid_motifsite_mat) %in% as.character(cpg_df_inmotif$position)] <- "CpG" -->
<!-- DT::datatable(rid_motifsite_mat,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','CpG positions in the motif for each read'),options = list(pageLength = 10) ) -->
<!-- ``` -->
<!-- We then retrieve methylation information from res$read. -->
<!-- ```{r} -->
<!-- # Fill methylation -->
<!-- reads_cpg <- res$reads[res$reads$base == "CG",] -->
<!-- reads_cpg$RID <- as.character(reads_cpg$RID) -->
<!-- idx <- which(reads_cpg$RID %in% rownames(rid_motifsite_methy) & -->
<!--                reads_cpg$ref_position %in% colnames(rid_motifsite_methy)) -->
<!-- for (j in idx) { -->
<!--   rid_motifsite_methy[reads_cpg$RID[j], as.character(reads_cpg$ref_position[j])] <- reads_cpg$mod_qual[j] -->
<!-- } -->
<!-- DT::datatable(rid_motifsite_methy,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation quality score for each read'),options = list(pageLength = 10) ) -->
<!-- stopifnot(identical(dimnames(rid_motifsite_methy), dimnames(rid_motifsite_mat))) -->
<!-- ``` -->
<!-- By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mCpG. A read is classified as mCpG if it contains at least one methylated CpG. -->
<!-- ```{r} -->
<!-- # Combine CpG & methylation -->
<!-- rid_motifsite_combined <- ifelse( -->
<!--   !is.na(rid_motifsite_methy) & !is.na(rid_motifsite_mat),  -->
<!--   "mCpG", -->
<!--   ifelse(!is.na(rid_motifsite_mat), "CpG", NA) -->
<!-- ) -->
<!-- DT::datatable(rid_motifsite_combined,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation annotation for each read'),options = list(pageLength = 10) ) -->
<!-- ``` -->
<!-- If a read has at least one mCpG, we annotate this read as mCpG -->
<!-- ```{r} -->
<!-- RID_cpg <- data.frame( -->
<!--   RID = rownames(rid_motifsite_combined), -->
<!--   status = apply(rid_motifsite_combined, 1, function(x) { -->
<!--     if ("mCpG" %in% x) "mCpG" -->
<!--     else if (all(na.omit(x) == "CpG")) "CpG" -->
<!--     else NA -->
<!--   }), -->
<!--   row.names = NULL -->
<!-- ) -->
<!-- DT::datatable(RID_cpg,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation info for each read'),options = list(pageLength = 10) ) -->
<!-- ``` -->
<!-- This results in two groups of reads: those containing mCpG and those without methylation. We then compare footprint frequencies between the two groups. -->
<!-- We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome. -->
<!-- ```{r} -->
<!-- fps <- res$fiberHMM_fps_infire -->
<!-- fps_TF <- fps[fps$class %in% c("< 40bp", "40-60bp"),] -->
<!-- fps_gr <- GRanges(fps_TF$chr, IRanges(fps_TF$start, fps_TF$end), strand=fps_TF$strand) -->
<!-- mcols(fps_gr) <- fps_TF[, !(names(fps_TF) %in% c("chr","start","end","strand"))] -->
<!-- hw <- GRanges(region$chr, IRanges(start_motif1, end_motif1)) -->
<!-- fps_gr <- fps_gr[width(pintersect(fps_gr, hw)) > 0] -->
<!-- rid_motif1_fp <- as.character(fps_gr$RID) -->
<!-- fps <- res$fiberHMM_fps -->
<!-- fps_nucl <- fps[fps$size %in% c("> 100bp", "60-100bp"),] -->
<!-- fps_gr_nucl <- GRanges(fps_nucl$chr, IRanges(fps_nucl$fp_blockStarts, fps_nucl$fp_blockEnds)) -->
<!-- mcols(fps_gr_nucl) <- fps_nucl[, !(names(fps_nucl) %in% c("chr","start","end"))] -->
<!-- fps_gr_nucl <- fps_gr_nucl[width(pintersect(fps_gr_nucl, hw, ignore.strand = T)) > 0] -->
<!-- rid_fps_nucl <- as.character(fps_gr_nucl$RID) -->
<!-- # filter rid sets -->
<!-- rid_motif1_without_nucl <- setdiff(rid_motif1, rid_fps_nucl) -->
<!-- RIDs_overlap_both <- intersect(rid_motif1_without_nucl, RID_cpg$RID) -->
<!-- RIDs_access.df <- data.frame( -->
<!--   RID = RIDs_overlap_both, -->
<!--   TF = RIDs_overlap_both %in% rid_motif1_fp, -->
<!--   mCpG = RIDs_overlap_both %in% RID_cpg$RID[RID_cpg$status=="mCpG"] -->
<!-- ) -->
<!-- RIDs_access.df$TF <- factor(RIDs_access.df$TF, levels=c(FALSE, TRUE)) -->
<!-- RIDs_access.df$mCpG <- factor(RIDs_access.df$mCpG, levels=c(FALSE, TRUE)) -->
<!-- print(RIDs_access.df) -->
<!-- contingency_table <- table(RIDs_access.df[,c("TF","mCpG")])  -->
<!-- prop_col <- round(prop.table(contingency_table, margin = 2) * 100,2) -->
<!-- print(contingency_table) -->
<!-- print(prop_col) -->
<!-- ``` -->
</div>
<div id="site27125" class="section level2">
<h2>site27125</h2>
<pre class="r"><code>site &lt;- &quot;site27125&quot;

index_region &lt;- which(singleton_motifs$name == site)

test_range_str &lt;- paste0(df_final$chr[index_region],&quot;:&quot;,df_final$start[index_region],&quot;-&quot;,df_final$end[index_region])
plot_name &lt;- paste0(TF1,&quot;_pairs_&quot;,sample_name,&quot;_&quot;,test_range_str)
result_dir &lt;- paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)

region &lt;- get_region(result_dir)
subtitle &lt;- sprintf(&quot;%s:%s-%s&quot;, region$chr, region$start, region$end)

# motif coordinates
start &lt;- singleton_motifs$start[index_region] + 100
end   &lt;- singleton_motifs$end[index_region] -100

region_plot &lt;- list(chr = df_final$chr[index_region],
                    start = start - 150,
                    end = end + 150)

res &lt;- try(get_region_result(result_dir,
                             filter_fire_reads = FALSE,
                             cover_region_only = FALSE,
                             select_A_only = FALSE,
                             split_haplotype = FALSE,
                             fiberHMM_size_breaks = c(60,100),
                             fiberHMM_cols = c(&quot;magenta&quot;,&quot;royalblue&quot;,&quot;darkgreen&quot;),
                             highlight_window = data.frame(start=start, end=end),
                             color_highlight_window = &quot;#44d6d8ff&quot;,
                             plot_name = plot_name,
                             # cluster = &quot;fiberHMM_configs&quot;,
                             # cluster_regions = cluster_regions,
                             show_RIDs = F,
                             region = region_plot,
                             subtitle = subtitle,
                             verbose = T)
)</code></pre>
<p>Get region result… Loading FiberHMM result from
/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_singleton//NRF1_pairs_LCL_18508_chr18:2906300-2907600/parsed
fiberHMM size levels: &lt; 60 bp 60-100 bp &gt; 100 bp Clustering reads…
36 reads. highlight windows: start end 1 2906988 2906999</p>
<pre class="r"><code>gg &lt;- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = &quot;v&quot;,
  axis = &quot;b&quot;
))</code></pre>
<p><img src="figure/TF_co-acc_NRF1_cpg2.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;We identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. \n&quot;)</code></pre>
<p>We identify CpG positions within this motif. The CpG positions are
extracted from the corresponding .fa files.</p>
<pre class="r"><code>cat(&quot;We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome&quot;)</code></pre>
<p>We restrict the analysis to reads without nucleosome footprints,
since transcription factors cannot bind when the motif is occupied by a
nucleosome</p>
<pre class="r"><code>DT::datatable(rid_motifsite_mat[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;CpG positions in the motif for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-b11be92bafaf5007f73e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b11be92bafaf5007f73e">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">CpG positions in the motif for each read --site27125<\/caption>","data":[["m64403e_220914_215953/71106831/ccs","m64403e_220914_215953/91881891/ccs","m64403e_220916_085351/105383918/ccs","m64403e_220916_085351/152174884/ccs","m64403e_220916_085351/25758340/ccs","m64403e_220916_085351/393431/ccs","m64403e_220916_085351/42860848/ccs","m64403e_230429_051256/104073307/ccs","m64403e_230429_051256/110756405/ccs","m64403e_230429_051256/111740283/ccs","m64403e_230429_051256/128713881/ccs","m64403e_230429_051256/130090317/ccs","m64403e_230429_051256/146671072/ccs","m64403e_230429_051256/159320443/ccs","m64403e_230429_051256/65537122/ccs","m64403e_230429_051256/66782722/ccs","m64403e_230429_051256/84476459/ccs","m64403e_230430_142929/177668396/ccs","m64403e_230430_142929/28116272/ccs","m64403e_230430_142929/37422008/ccs","m64403e_230430_142929/71958806/ccs","m64403e_230430_142929/77793541/ccs","m64403e_230430_142929/98960949/ccs","m64403e_230501_235831/152898117/ccs","m64403e_230501_235831/156763069/ccs","m64403e_230501_235831/169215688/ccs","m64403e_230501_235831/179046485/ccs","m64403e_230501_235831/24185018/ccs","m64403e_230501_235831/52429822/ccs","m64403e_230501_235831/53348883/ccs","m64403e_230501_235831/54856338/ccs","m64403e_230501_235831/59443299/ccs","m64403e_230501_235831/73533218/ccs","m64403e_230501_235831/84805252/ccs","m64403e_230501_235831/94306929/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],["C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>2906988<\/th>\n      <th>2906989<\/th>\n      <th>2906990<\/th>\n      <th>2906991<\/th>\n      <th>2906992<\/th>\n      <th>2906993<\/th>\n      <th>2906994<\/th>\n      <th>2906995<\/th>\n      <th>2906996<\/th>\n      <th>2906997<\/th>\n      <th>2906998<\/th>\n      <th>2906999<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;We then retrieve methylation information from res$read. \n&quot;)</code></pre>
<p>We then retrieve methylation information from res$read.</p>
<pre class="r"><code>DT::datatable(rid_motifsite_methy[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation quality score for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-bc27f7549192058b9f6a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bc27f7549192058b9f6a">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation quality score for each read --site27125<\/caption>","data":[["m64403e_220914_215953/71106831/ccs","m64403e_220914_215953/91881891/ccs","m64403e_220916_085351/105383918/ccs","m64403e_220916_085351/152174884/ccs","m64403e_220916_085351/25758340/ccs","m64403e_220916_085351/393431/ccs","m64403e_220916_085351/42860848/ccs","m64403e_230429_051256/104073307/ccs","m64403e_230429_051256/110756405/ccs","m64403e_230429_051256/111740283/ccs","m64403e_230429_051256/128713881/ccs","m64403e_230429_051256/130090317/ccs","m64403e_230429_051256/146671072/ccs","m64403e_230429_051256/159320443/ccs","m64403e_230429_051256/65537122/ccs","m64403e_230429_051256/66782722/ccs","m64403e_230429_051256/84476459/ccs","m64403e_230430_142929/177668396/ccs","m64403e_230430_142929/28116272/ccs","m64403e_230430_142929/37422008/ccs","m64403e_230430_142929/71958806/ccs","m64403e_230430_142929/77793541/ccs","m64403e_230430_142929/98960949/ccs","m64403e_230501_235831/152898117/ccs","m64403e_230501_235831/156763069/ccs","m64403e_230501_235831/169215688/ccs","m64403e_230501_235831/179046485/ccs","m64403e_230501_235831/24185018/ccs","m64403e_230501_235831/52429822/ccs","m64403e_230501_235831/53348883/ccs","m64403e_230501_235831/54856338/ccs","m64403e_230501_235831/59443299/ccs","m64403e_230501_235831/73533218/ccs","m64403e_230501_235831/84805252/ccs","m64403e_230501_235831/94306929/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,0.6933594,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.8964844,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,0.7363281,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,0.9082031,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.7480469,null,null,null,0.6933594,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,0.7363281,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>2906988<\/th>\n      <th>2906989<\/th>\n      <th>2906990<\/th>\n      <th>2906991<\/th>\n      <th>2906992<\/th>\n      <th>2906993<\/th>\n      <th>2906994<\/th>\n      <th>2906995<\/th>\n      <th>2906996<\/th>\n      <th>2906997<\/th>\n      <th>2906998<\/th>\n      <th>2906999<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mC.\n&quot;)</code></pre>
<p>By combining the CpG position matrix with the methylation quality
matrix, we annotate methylated sites as mC.</p>
<pre class="r"><code>DT::datatable(rid_motifsite_combined[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation annotation for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-8b2d970f83f6c5ce0219" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8b2d970f83f6c5ce0219">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation annotation for each read --site27125<\/caption>","data":[["m64403e_220914_215953/71106831/ccs","m64403e_220914_215953/91881891/ccs","m64403e_220916_085351/105383918/ccs","m64403e_220916_085351/152174884/ccs","m64403e_220916_085351/25758340/ccs","m64403e_220916_085351/393431/ccs","m64403e_220916_085351/42860848/ccs","m64403e_230429_051256/104073307/ccs","m64403e_230429_051256/110756405/ccs","m64403e_230429_051256/111740283/ccs","m64403e_230429_051256/128713881/ccs","m64403e_230429_051256/130090317/ccs","m64403e_230429_051256/146671072/ccs","m64403e_230429_051256/159320443/ccs","m64403e_230429_051256/65537122/ccs","m64403e_230429_051256/66782722/ccs","m64403e_230429_051256/84476459/ccs","m64403e_230430_142929/177668396/ccs","m64403e_230430_142929/28116272/ccs","m64403e_230430_142929/37422008/ccs","m64403e_230430_142929/71958806/ccs","m64403e_230430_142929/77793541/ccs","m64403e_230430_142929/98960949/ccs","m64403e_230501_235831/152898117/ccs","m64403e_230501_235831/156763069/ccs","m64403e_230501_235831/169215688/ccs","m64403e_230501_235831/179046485/ccs","m64403e_230501_235831/24185018/ccs","m64403e_230501_235831/52429822/ccs","m64403e_230501_235831/53348883/ccs","m64403e_230501_235831/54856338/ccs","m64403e_230501_235831/59443299/ccs","m64403e_230501_235831/73533218/ccs","m64403e_230501_235831/84805252/ccs","m64403e_230501_235831/94306929/ccs"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],["C","C","mC","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","mC","C","C","C","C","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G","G"],["C","C","mC","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","mC","C","C","C","mC","C","C","C","C","C","C","C","C","C"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>2906988<\/th>\n      <th>2906989<\/th>\n      <th>2906990<\/th>\n      <th>2906991<\/th>\n      <th>2906992<\/th>\n      <th>2906993<\/th>\n      <th>2906994<\/th>\n      <th>2906995<\/th>\n      <th>2906996<\/th>\n      <th>2906997<\/th>\n      <th>2906998<\/th>\n      <th>2906999<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;We summarize the methylation and TF footprint status for each read\n&quot;)</code></pre>
<p>We summarize the methylation and TF footprint status for each
read</p>
<pre class="r"><code>DT::datatable(RIDs_access.df_all[[site]],caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&#39;Methylation and TF footprint status for each read --&#39;,site)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-979009f5d5c312be05de" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-979009f5d5c312be05de">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Methylation and TF footprint status for each read --site27125<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],["m64403e_220916_085351/105383918/ccs","m64403e_220916_085351/25758340/ccs","m64403e_220916_085351/393431/ccs","m64403e_230429_051256/104073307/ccs","m64403e_230429_051256/110756405/ccs","m64403e_230429_051256/130090317/ccs","m64403e_230429_051256/146671072/ccs","m64403e_230429_051256/159320443/ccs","m64403e_230429_051256/66782722/ccs","m64403e_230429_051256/84476459/ccs","m64403e_230501_235831/156763069/ccs","m64403e_230501_235831/24185018/ccs","m64403e_230501_235831/59443299/ccs","m64403e_230501_235831/84805252/ccs"],["FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE"],["TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE"],["TRUE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE","FALSE"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>RID<\/th>\n      <th>TF<\/th>\n      <th>pos1<\/th>\n      <th>pos2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cat(&quot;For position 1, we have\n&quot;)</code></pre>
<p>For position 1, we have</p>
<pre class="r"><code>kable(as.matrix(contingency_table_pos1_all[[site]]),
      caption = paste(&quot;Contingency table for&quot;, site))</code></pre>
<table>
<caption>Contingency table for site27125</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">13</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;The proportion of TF footprint in different methylation groups&quot;)</code></pre>
<p>The proportion of TF footprint in different methylation groups</p>
<pre class="r"><code>kable(as.matrix(prop_col_pos1_all[[site]]),
      caption = paste(&quot;Proportion table for&quot;, site))</code></pre>
<table>
<caption>Proportion table for site27125</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">100</td>
<td align="right">100</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;For position 2, we have\n&quot;)</code></pre>
<p>For position 2, we have</p>
<pre class="r"><code>kable(as.matrix(contingency_table_pos2_all[[site]]),
      caption = paste(&quot;Contingency table for&quot;, site))</code></pre>
<table>
<caption>Contingency table for site27125</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos1_FALSE</th>
<th align="right">CpG_pos1_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">13</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(&quot;The proportion of TF footprint in different methylation groups&quot;)</code></pre>
<p>The proportion of TF footprint in different methylation groups</p>
<pre class="r"><code>kable(as.matrix(prop_col_pos2_all[[site]]),
      caption = paste(&quot;Proportion table for&quot;, site))</code></pre>
<table>
<caption>Proportion table for site27125</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CpG_pos2_FALSE</th>
<th align="right">CpG_pos2_TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TF_FALSE</td>
<td align="right">100</td>
<td align="right">100</td>
</tr>
<tr class="even">
<td align="left">TF_TRUE</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="summary-for-each-methylation-group" class="section level1">
<h1>Summary for each methylation group</h1>
<p>We only consider the motifs with at least one CpGm read.</p>
<p>We did paired t-test to test if the footprint detected proportion are
different.</p>
<pre class="r"><code>contingency_prop_col &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/contingency_prop_col_pos1_sites_select.RDS&quot;)

# extract the 2nd row of each table if it&#39;s valid
mat &lt;- contingency_prop_col %&gt;%
  imap_dfr(~ {
    if (all(is.nan(.x[,2]))) return(NULL)
    tibble(
      methylated     = .x[2, 1],
      not_methylated = .x[2, 2],
      motif          = .y
    )
  })

# run paired t-test
t &lt;- t.test(mat$methylated, mat$not_methylated, paired = TRUE)
pval_label &lt;- paste0(&quot;paired t-test p = \n&quot;, signif(t$p.value, 3))

# long format for plotting
df_long &lt;- mat %&gt;%
  pivot_longer(cols = c(methylated, not_methylated),
               names_to = &quot;CpG&quot;, 
               values_to = &quot;% of Footprint&quot;)

# plot
p1 &lt;- ggplot(df_long, aes(x = CpG, y = `% of Footprint`, fill = CpG)) +
  geom_boxplot() +
  theme_minimal() +
  ggtitle(&quot;Position 1&quot;) +
  annotate(&quot;text&quot;, x = 1.5,
           y = max(df_long$`% of Footprint`, na.rm = TRUE) * 1.05,
           label = pval_label)

contingency_prop_col &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/contingency_prop_col_pos2_sites_select.RDS&quot;)

# extract the 2nd row of each table if it&#39;s valid
mat &lt;- contingency_prop_col %&gt;%
  imap_dfr(~ {
    if (all(is.nan(.x[,2]))) return(NULL)
    tibble(
      methylated     = .x[2, 1],
      not_methylated = .x[2, 2],
      motif          = .y
    )
  })

# run paired t-test
t &lt;- t.test(mat$methylated, mat$not_methylated, paired = TRUE)
pval_label &lt;- paste0(&quot;paired t-test p = \n&quot;, signif(t$p.value, 3))

# long format for plotting
df_long &lt;- mat %&gt;%
  pivot_longer(cols = c(methylated, not_methylated),
               names_to = &quot;CpG&quot;, 
               values_to = &quot;% of Footprint&quot;)

# plot
p2 &lt;- ggplot(df_long, aes(x = CpG, y = `% of Footprint`, fill = CpG)) +
  geom_boxplot() +
  theme_minimal() +
  ggtitle(&quot;Position 2&quot;) +
  annotate(&quot;text&quot;, x = 1.5,
           y = max(df_long$`% of Footprint`, na.rm = TRUE) * 1.05,
           label = pval_label)


grid.arrange(p1, p2, ncol = 2)</code></pre>
<p><img src="figure/TF_co-acc_NRF1_cpg2.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># contingency_table &lt;- readRDS(&quot;/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_singleton_cpg/contingency_table_pos1_sites_select.RDS&quot;)
# 
# mat_table &lt;- c()
# mat_table_nfp &lt;- c()
# names &lt;- c()
# for (i in 1:length(contingency_prop_col)){
#   
#   table &lt;- contingency_table[[i]]
#   
#   if(is.null(table) || (length(table) == 1 &amp;&amp; is.na(table))) next
#   
#   extract &lt;- contingency_table[[i]][2,]
#   names &lt;- c(names,names(contingency_table[i]))
#   mat_table &lt;- rbind(mat_table,extract)
#   mat_table_nfp &lt;- rbind(mat_table_nfp,contingency_table[[i]][1,])
#   
# }
# 
# 
# mat_table &lt;- as.data.frame(mat_table)
# rownames(mat_table) &lt;- NULL
# colnames(mat_table) &lt;- c(&quot;methylated&quot;,&quot;not_methylated&quot;)
# mat_table$motif &lt;- names
# 
# mat_table_nfp &lt;- as.data.frame(mat_table_nfp)
# rownames(mat_table_nfp) &lt;- NULL
# colnames(mat_table_nfp) &lt;- c(&quot;methylated&quot;,&quot;not_methylated&quot;)
# mat_table_nfp$motif &lt;- names
# 
# 
# tmp &lt;- merge(mat_table, mat_table_nfp, by = &quot;motif&quot;)
# combine &lt;- merge(tmp, mat, by = &quot;motif&quot;)
# 
# colnames(combine) &lt;- c(&quot;motifsite&quot;,&quot;#of_fp_detected_in_methy_group&quot;,&quot;#of_fp_detected_in_un_methy_group&quot;,&quot;#of_no_fp_reads_in_methy_group&quot;,&quot;#of_no_fp_reads_in_un_methy_group&quot;,&quot;%of_fp_detected_in_methy_group&quot;,&quot;%of_fp_detected_in_un_methy_group&quot;)
# 
# DT::datatable(combine,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;footprint detected, %of_fp_detected_in_methy_group is calculated as: #of_fp_detected_in_methy_group/(#of_fp_detected_in_methy_group+#of_fp_detected_in_un_methy_group)\n&#39;),options = list(pageLength = 10) )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] Biostrings_2.64.0    XVector_0.36.0       gridExtra_2.3       
 [4] forcats_0.5.1        stringr_1.5.1        dplyr_1.1.4         
 [7] purrr_1.0.2          readr_2.1.2          tidyr_1.3.0         
[10] tibble_3.2.1         tidyverse_1.3.1      ggplot2_3.5.1       
[13] GenomicRanges_1.48.0 GenomeInfoDb_1.39.9  IRanges_2.30.0      
[16] S4Vectors_0.34.0     BiocGenerics_0.42.0  reshape2_1.4.4      
[19] data.table_1.14.2    knitr_1.39          

loaded via a namespace (and not attached):
 [1] bitops_1.0-7           fs_1.5.2               lubridate_1.8.0       
 [4] httr_1.4.3             rprojroot_2.0.3        tools_4.2.0           
 [7] backports_1.4.1        bslib_0.3.1            utf8_1.2.2            
[10] R6_2.5.1               DT_0.22                DBI_1.2.2             
[13] colorspace_2.0-3       withr_2.5.0            tidyselect_1.2.0      
[16] compiler_4.2.0         git2r_0.30.1           cli_3.6.1             
[19] rvest_1.0.2            xml2_1.3.3             labeling_0.4.2        
[22] sass_0.4.1             scales_1.3.0           digest_0.6.29         
[25] rmarkdown_2.25         pkgconfig_2.0.3        htmltools_0.5.2       
[28] highr_0.9              dbplyr_2.1.1           fastmap_1.1.0         
[31] htmlwidgets_1.5.4      rlang_1.1.2            readxl_1.4.0          
[34] rstudioapi_0.13        jquerylib_0.1.4        generics_0.1.2        
[37] farver_2.1.0           jsonlite_1.8.0         crosstalk_1.2.0       
[40] RCurl_1.98-1.7         magrittr_2.0.3         GenomeInfoDbData_1.2.8
[43] Rcpp_1.0.12            munsell_0.5.0          fansi_1.0.3           
[46] lifecycle_1.0.4        stringi_1.7.6          whisker_0.4           
[49] yaml_2.3.5             zlibbioc_1.42.0        plyr_1.8.7            
[52] grid_4.2.0             promises_1.2.0.1       crayon_1.5.1          
[55] haven_2.5.0            cowplot_1.1.1          hms_1.1.1             
[58] pillar_1.9.0           reprex_2.0.1           glue_1.6.2            
[61] evaluate_0.15          modelr_0.1.8           vctrs_0.6.5           
[64] tzdb_0.4.0             httpuv_1.6.5           cellranger_1.1.0      
[67] gtable_0.3.0           assertthat_0.2.1       xfun_0.41             
[70] broom_0.8.0            later_1.3.0            workflowr_1.7.0       
[73] ellipsis_0.3.2        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
