<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-08-31" />

<title>TF Co-accessibility for NRF1 &amp; SP1</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiberseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/fiberseq">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">TF Co-accessibility for NRF1 &amp; SP1</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-08-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>fiberseq/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250831code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250831)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250831code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250831)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project/xinhe/xsun/website/fiberseq/data/example_res_from_plot_region_reads.RDS
</td>
<td style="text-align:left;">
data/example_res_from_plot_region_reads.RDS
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229fiberseqtree4b2f0bdbe2c1804c3f353546ee763aa75a20c134targetblank4b2f0bda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/fiberseq/tree/4b2f0bdbe2c1804c3f353546ee763aa75a20c134" target="_blank">4b2f0bd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229fiberseqtree4b2f0bdbe2c1804c3f353546ee763aa75a20c134targetblank4b2f0bda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/fiberseq/tree/4b2f0bdbe2c1804c3f353546ee763aa75a20c134" target="_blank">4b2f0bd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/TF_co-acc_NRF1_SP1_400800apart.Rmd

Unstaged changes:
    Modified:   analysis/TF_co-acc_NRF1_SP1.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/TF_co-acc_NRF1_SP1.Rmd</code>)
and HTML (<code>docs/TF_co-acc_NRF1_SP1.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/4b2f0bdbe2c1804c3f353546ee763aa75a20c134/analysis/TF_co-acc_NRF1_SP1.Rmd" target="_blank">4b2f0bd</a>
</td>
<td>
XSun
</td>
<td>
2025-09-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/fiberseq/4b2f0bdbe2c1804c3f353546ee763aa75a20c134/docs/TF_co-acc_NRF1_SP1.html" target="_blank">4b2f0bd</a>
</td>
<td>
XSun
</td>
<td>
2025-09-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/analysis/TF_co-acc_NRF1_SP1.Rmd" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/fiberseq/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/TF_co-acc_NRF1_SP1.html" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/cb3361cf5581d1737d73e80e1d512d85ffa40c74/analysis/TF_co-acc_NRF1_SP1.Rmd" target="_blank">cb3361c</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/fiberseq/cb3361cf5581d1737d73e80e1d512d85ffa40c74/docs/TF_co-acc_NRF1_SP1.html" target="_blank">cb3361c</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(data.table)
library(reshape2)
library(GenomicRanges)
library(ggplot2)
library(tidyverse)

source(&quot;/project/spott/kevinluo/Fiber_seq/fiberhub/code/process_fiberseq_data.R&quot;)
source(&quot;/project/spott/kevinluo/Fiber_seq/fiberhub/code/plots.R&quot;)

TF1 &lt;- &quot;NRF1&quot;
TF2 &lt;- &quot;SP1&quot;</code></pre>
<div id="motifs-are-50-200bp-apart" class="section level1">
<h1>Motifs are 50-200bp apart</h1>
<div id="extract-the-regions-containing-both-tf-motif-sites"
class="section level2">
<h2>Extract the regions containing both TF motif sites</h2>
<p>Regions containing both TFs were extracted, and only those in which
the TFs were separated by more than <code>50 bp</code> but less than
<code>200 bp</code> were retained. Each motif was then extended by 500
bp on both sides to define the final regions.</p>
<pre class="r"><code>motif_sites_motif1 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF1,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif1_chip1 &lt;- motif_sites_motif1[motif_sites_motif1$chip_label == 1,]

motif_sites_motif2 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF2,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif2_chip1 &lt;- motif_sites_motif1[motif_sites_motif2$chip_label == 1,]

motif1 &lt;- motif_sites_motif1_chip1
motif1 &lt;- motif1[!is.na(motif1$start),]
motif2 &lt;- motif_sites_motif2_chip1
motif2 &lt;- motif2[!is.na(motif2$start),]

## Convert to GRanges
gr_motif1 &lt;- GRanges(seqnames = motif1$chr,
                 ranges   = IRanges(start = motif1$start+100, end = motif1$end-100),
                 site     = motif1$name)

gr_motif2 &lt;- GRanges(seqnames = motif2$chr,
                 ranges   = IRanges(start = motif2$start+100, end = motif2$end-100),
                 site     = motif2$name)

## Define distance cutoff
dist_cutoff_large &lt;- 200
dist_cutoff_small &lt;- 50

## Find nearest motif2 site for each motif1 site
nearest_hits &lt;- distanceToNearest(gr_motif1, gr_motif2)

## Filter by distance threshold
close_pairs &lt;- nearest_hits[mcols(nearest_hits)$distance &lt;= dist_cutoff_large &amp; mcols(nearest_hits)$distance &gt;= dist_cutoff_small]

## Extract regions
motif1_close &lt;- gr_motif1[queryHits(close_pairs)]
motif2_close &lt;- gr_motif2[subjectHits(close_pairs)]

## Make a data.frame
df_close &lt;- data.frame(
  chr = as.character(seqnames(motif1_close)),
  motif1_start = start(motif1_close),
  motif1_end   = end(motif1_close),
  motif1_site  = motif1_close$site,
  motif2_start  = start(motif2_close),
  motif2_end    = end(motif2_close),
  motif2_site   = motif2_close$site,
  distance   = mcols(close_pairs)$distance
)

DT::datatable(df_close,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Motif sites selected&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-55d484e2ce688c3024c2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-55d484e2ce688c3024c2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Motif sites selected<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68"],["chr1","chr1","chr1","chr2","chr3","chr4","chr4","chr4","chr4","chr4","chr4","chr5","chr7","chr7","chr7","chr8","chr8","chr8","chr8","chr8","chr8","chr9","chr9","chr9","chr9","chr10","chr10","chr11","chr12","chr12","chr12","chr12","chr13","chr13","chr14","chr15","chr15","chr15","chr15","chr16","chr16","chr16","chr17","chr17","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr20","chr20","chr20","chr20","chr21","chr21","chr22","chr22","chr22"],[1308519,1308520,234479110,213151684,31532305,1248717,1248718,1248779,1248780,128288216,128288217,171387600,100429698,112391830,112391934,42391690,42391691,108248849,144428635,144428640,144428641,6758068,6758069,77177509,122159829,104122018,104122019,65570030,120116857,120116858,123533763,123533764,76991992,76992033,22589374,63600915,63600989,63600990,65530394,50065908,67481299,67481300,49362376,49362377,4400621,4400622,10928619,10928620,10960853,10960854,16628475,55600212,56120867,56120868,56120896,56120897,56120976,56120977,56404077,62796386,62796387,64079855,64079856,33324872,33324873,24011237,24011238,27919446],[1308530,1308531,234479121,213151695,31532316,1248728,1248729,1248790,1248791,128288227,128288228,171387611,100429709,112391841,112391945,42391701,42391702,108248860,144428646,144428651,144428652,6758079,6758080,77177520,122159840,104122029,104122030,65570041,120116868,120116869,123533774,123533775,76992003,76992044,22589385,63600926,63601000,63601001,65530405,50065919,67481310,67481311,49362387,49362388,4400632,4400633,10928630,10928631,10960864,10960865,16628486,55600223,56120878,56120879,56120907,56120908,56120987,56120988,56404088,62796397,62796398,64079866,64079867,33324883,33324884,24011248,24011249,27919457],["site126","site127","site2731","site4794","site5576","site7065","site7066","site7067","site7068","site7994","site7995","site9922","site12629","site12798","site12799","site13798","site13799","site14228","site14599","site14600","site14601","site14703","site14704","site15202","site15684","site17254","site17255","site18586","site20631","site20632","site20719","site20720","site21316","site21317","site21776","site23214","site23215","site23216","site23243","site24601","site24845","site24846","site26395","site26396","site28273","site28274","site28580","site28581","site28592","site28593","site28779","site29970","site30004","site30005","site30006","site30007","site30008","site30009","site30028","site31107","site31108","site31189","site31190","site31605","site31606","site32252","site32253","site32307"],[1308583,1308583,234478936,213151856,31532093,1248895,1248895,1248895,1248895,128288130,128288130,171387724,100429785,112391726,112391726,42391823,42391823,108248779,144428568,144428568,144428568,6757950,6757950,77177431,122159766,104121886,104121886,65570096,120116920,120116920,123533635,123533635,76992144,76992144,22589281,63600780,63600780,63600780,65530467,50065983,67481197,67481197,49362265,49362265,4400525,4400525,10928736,10928736,10961031,10961031,16628351,55600139,56121042,56121042,56121042,56121042,56121042,56121042,56404015,62796454,62796454,64079961,64079961,33325008,33325008,24011170,24011170,27919257],[1308594,1308594,234478947,213151867,31532104,1248906,1248906,1248906,1248906,128288141,128288141,171387735,100429796,112391737,112391737,42391834,42391834,108248790,144428579,144428579,144428579,6757961,6757961,77177442,122159777,104121897,104121897,65570107,120116931,120116931,123533646,123533646,76992155,76992155,22589292,63600791,63600791,63600791,65530478,50065994,67481208,67481208,49362276,49362276,4400536,4400536,10928747,10928747,10961042,10961042,16628362,55600150,56121053,56121053,56121053,56121053,56121053,56121053,56404026,62796465,62796465,64079972,64079972,33325019,33325019,24011181,24011181,27919268],["site129","site129","site2730","site4797","site5574","site7070","site7070","site7070","site7070","site7993","site7993","site9923","site12630","site12796","site12796","site13802","site13802","site14227","site14598","site14598","site14598","site14702","site14702","site15201","site15680","site17251","site17251","site18587","site20634","site20634","site20718","site20718","site21319","site21319","site21774","site23213","site23213","site23213","site23244","site24603","site24844","site24844","site26393","site26393","site28272","site28272","site28582","site28582","site28596","site28596","site28777","site29969","site30011","site30011","site30011","site30011","site30011","site30011","site30025","site31109","site31109","site31191","site31191","site31608","site31608","site32249","site32249","site32306"],[52,51,162,160,200,166,165,104,103,74,75,112,75,92,196,121,120,58,55,60,61,106,107,66,51,120,121,54,51,50,116,117,140,99,81,123,197,198,61,63,90,91,99,100,84,85,105,104,166,165,112,61,163,162,134,133,54,53,50,56,55,94,93,124,123,55,56,177]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>motif1_start<\/th>\n      <th>motif1_end<\/th>\n      <th>motif1_site<\/th>\n      <th>motif2_start<\/th>\n      <th>motif2_end<\/th>\n      <th>motif2_site<\/th>\n      <th>distance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_summary &lt;- df_close %&gt;%
  group_by(chr, motif2_site, motif2_start, motif2_end) %&gt;%
  summarise(
    min_motif1_start = min(motif1_start),
    max_motif1_end   = max(motif1_end),
    .groups = &quot;drop&quot;
  )

df_final &lt;- df_summary %&gt;%
  transmute(
    chr,
    start = pmin(motif2_start, min_motif1_start) - 500,
    end   = pmax(motif2_end,   max_motif1_end)   + 500
  )

df_final &lt;- df_final %&gt;%
  mutate(chr_num = as.numeric(gsub(&quot;chr&quot;, &quot;&quot;, chr))) %&gt;%
  arrange(chr_num, start) %&gt;%
  select(-chr_num)

DT::datatable(df_final,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Regions selected for plotting &#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-0c375f72d786c94bb27a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c375f72d786c94bb27a">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Regions selected for plotting <\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38"],["chr1","chr1","chr2","chr3","chr4","chr4","chr5","chr7","chr7","chr8","chr8","chr8","chr9","chr9","chr9","chr10","chr11","chr12","chr12","chr13","chr14","chr15","chr15","chr16","chr16","chr17","chr19","chr19","chr19","chr19","chr19","chr19","chr19","chr20","chr20","chr21","chr22","chr22"],[1308019,234478436,213151184,31531593,1248217,128287630,171387100,100429198,112391226,42391190,108248279,144428068,6757450,77176931,122159266,104121386,65569530,120116357,123533135,76991492,22588781,63600280,65529894,50065408,67480697,49361765,4400025,10928119,10960353,16627851,55599639,56120367,56403515,62795886,64079355,33324372,24010670,27918757],[1309094,234479621,213152367,31532816,1249406,128288728,171388235,100430296,112392445,42392334,108249360,144429152,6758580,77178020,122160340,104122530,65570607,120117431,123534275,76992655,22589885,63601501,65530978,50066494,67481811,49362888,4401133,10929247,10961542,16628986,55600723,56121553,56404588,62796965,64080472,33325519,24011749,27919957]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualization-and-co-accessibility-test"
class="section level2">
<h2>Visualization and co-accessibility test</h2>
<p>For visualization, highlighted windows are the ones
<code>±7 bp</code> around each motif.</p>
<p>To determine whether a TF has a footprint, footprint data were
extracted from <code>res$fiberHMM_fps_infire</code>.</p>
<p>Footprints were retained if they (1) had a FiberHMM footprint size
&lt;40 bp and (2) overlapped with the highlighted motif windows.</p>
<p>Next, reads covering both motif regions were extracted, and the
presence or absence of footprints for the two TFs was counted.</p>
<p>Last, we did FET</p>
<pre class="r"><code>res &lt;- readRDS(&quot;/project/xinhe/xsun/website/fiberseq/data/example_res_from_plot_region_reads.RDS&quot;)

head(res$fiberHMM_fps_infire)</code></pre>
<pre><code># A tibble: 6 x 10
  chr      start      end strand RID       class    ln read_start read_end tag  
  &lt;fct&gt;    &lt;int&gt;    &lt;int&gt; &lt;fct&gt;  &lt;fct&gt;     &lt;fct&gt; &lt;int&gt;      &lt;int&gt;    &lt;int&gt; &lt;chr&gt;
1 chr22 27919253 27919290 *      m64403e_~ &lt; 40~    37   27918289 27929700 H1   
2 chr22 27919394 27919417 *      m64403e_~ &lt; 40~    23   27918289 27929700 H1   
3 chr22 27919440 27919463 *      m64403e_~ &lt; 40~    23   27918289 27929700 H1   
4 chr22 27919526 27919548 *      m64403e_~ &lt; 40~    22   27918289 27929700 H1   
5 chr22 27925985 27926006 *      m64403e_~ &lt; 40~    21   27918289 27929700 H1   
6 chr22 27926153 27926173 *      m64403e_~ &lt; 40~    20   27918289 27929700 H1   </code></pre>
<pre class="r"><code>out_dir=&quot;/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results&quot;
sample_name=&quot;LCL_18508&quot;

df_close_nodup &lt;- df_close[!duplicated(df_close$motif2_site),]

#for (i in 1:3){
for (i in 1:nrow(df_final)){
  
  test_range_str &lt;- paste0(df_final$chr[i],&quot;:&quot;,df_final$start[i],&quot;-&quot;,df_final$end[i])
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)
  
  region &lt;- get_region(result_dir)
  subtitle &lt;- sprintf(&quot;%s:%s-%s&quot;, region$chr, region$start, region$end)
  
  print(subtitle)
  
  df_close_region &lt;- df_close[df_close$motif2_site == df_close_nodup$motif2_site[i],]
  min_start_motif1 &lt;- min(df_close_region$motif1_start)
  max_end_motif1   &lt;- max(df_close_region$motif1_end)
  
  highlight_windows.df &lt;- data.frame(start = c(min_start_motif1 - 7,df_close_nodup$motif2_start[i] -7 ), 
                              end = c(max_end_motif1 + 7 ,df_close_nodup$motif2_end[i] +7))
  
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)

  
  res &lt;- plot_region_reads(result_dir, 
                         filter_fire_reads = FALSE,
                         cover_region_only = FALSE,
                         cluster_reads = TRUE,
                         select_A_only = TRUE,
                         split_haplotype = FALSE,
                         highlight_window = highlight_windows.df,
                         color_highlight_window = &quot;#ade6b3&quot;,
                         plot_name = plot_name,
                         subtitle = subtitle)

  region &lt;- res$region
  window_size &lt;- res$window_size
  rids_df &lt;- res$rids_df
  pileup &lt;- res$pileup
  reads &lt;- res$reads
  fire &lt;- res$fire
  fdr_fire &lt;- res$fdr_fire
  fiberHMM_fps &lt;- res$fiberHMM_fps
  gg &lt;- res$gg
  
  
  print(cowplot::plot_grid(
    gg$pileup,
    gg$dimelo_reads,
    gg$fire_fdr,
    gg$fire_reads,
    gg$fiberHMM,
    ncol = 1,
    rel_heights = c(4,8,2,8,8),
    align = &quot;v&quot;,
    axis = &quot;b&quot;
  ))

  
  fps &lt;- res$fiberHMM_fps_infire
  fps_rid &lt;- fps$RID
  
  #fps_highlight_motif1 &lt;- fps[fps$start &gt; highlight_windows.df$start[1] &amp; fps$end &lt; highlight_windows.df$end[1], ]
  fps_highlight_motif1 &lt;- fps[fps$start &lt;= highlight_windows.df$end[1] &amp; fps$end &gt;= highlight_windows.df$start[1], ]
  fps_tf_motif1 &lt;- fps_highlight_motif1[fps_highlight_motif1$class == &quot;&lt; 40bp&quot;,]
  rid_motif1 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= max_end_motif1 &amp; res$rids_df$end &gt;= min_start_motif1])
  
  
  #fps_highlight_motif2 &lt;- fps[fps$start &gt; highlight_windows.df$start[2] &amp; fps$end &lt; highlight_windows.df$end[2], ]
  fps_highlight_motif2 &lt;- fps[fps$start &lt;= highlight_windows.df$end[2] &amp; fps$end &gt;= highlight_windows.df$start[2], ]
  fps_tf_motif2 &lt;- fps_highlight_motif2[fps_highlight_motif2$class == &quot;&lt; 40bp&quot;,]
  rid_motif2 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= df_close_nodup$motif2_end[i] &amp; res$rids_df$end &gt;= df_close_nodup$motif2_start[i]])
  
  RIDs_overlap_both_regions &lt;- intersect(rid_motif1, rid_motif2)
  
  RIDs_access.df &lt;- data.frame(RID = RIDs_overlap_both_regions, 
                               TF1 = FALSE,
                               TF2 = FALSE)
  
  RIDs_access.df$TF1[RIDs_access.df$RID %in% unique(fps_tf_motif1$RID)] &lt;- TRUE
  RIDs_access.df$TF2[RIDs_access.df$RID %in% unique(fps_tf_motif2$RID)] &lt;- TRUE
  
  
  # print(RIDs_access.df)
  
  if(!(sum(RIDs_access.df$TF1) &gt; 0 &amp; sum(RIDs_access.df$TF2) &gt; 0)) next 
  
  cat(&quot;Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...\n&quot;)
  print(contingency_table &lt;- table(RIDs_access.df[,c(&quot;TF1&quot;, &quot;TF2&quot;)]))
  
  print(fisher_res &lt;- fisher.test(contingency_table))
  
}</code></pre>
<pre><code>[1] &quot;chr1:1308018-1309094&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
41 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-1.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    31    1
  TRUE      8    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.0000 155.6231
sample estimates:
odds ratio 
         0 

[1] &quot;chr1:234478435-234479621&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-2.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr2:213151183-213152367&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
31 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-3">
Past versions of unnamed-chunk-4-3.png
</button>
</p>
<div id="fig-unnamed-chunk-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-3.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    28    1
  TRUE      1    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
    0.000 1111.377
sample estimates:
odds ratio 
         0 

[1] &quot;chr3:31531592-31532816&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
31 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-4">
Past versions of unnamed-chunk-4-4.png
</button>
</p>
<div id="fig-unnamed-chunk-4-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-4.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    14   12
  TRUE      0    2

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.4815
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1901047       Inf
sample estimates:
odds ratio 
       Inf 

[1] &quot;chr4:1248216-1249406&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
32 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-5">
Past versions of unnamed-chunk-4-5.png
</button>
</p>
<div id="fig-unnamed-chunk-4-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-5.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    11    0
  TRUE     18    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.0148602       Inf
sample estimates:
odds ratio 
       Inf 

[1] &quot;chr4:128287628-128288728&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
31 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-6.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-6">
Past versions of unnamed-chunk-4-6.png
</button>
</p>
<div id="fig-unnamed-chunk-4-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-6.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     2    5
  TRUE      7   14

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.0618476 6.6934454
sample estimates:
odds ratio 
 0.8062693 

[1] &quot;chr5:171387099-171388235&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
32 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-7.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-7">
Past versions of unnamed-chunk-4-7.png
</button>
</p>
<div id="fig-unnamed-chunk-4-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-7.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     9   10
  TRUE      3    8

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.4425
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.3908593 17.9557134
sample estimates:
odds ratio 
  2.331428 

[1] &quot;chr7:100429196-100430296&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-8.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-8">
Past versions of unnamed-chunk-4-8.png
</button>
</p>
<div id="fig-unnamed-chunk-4-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-8.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    21    7
  TRUE      1    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.469
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.03347547 245.90394776
sample estimates:
odds ratio 
  2.872491 

[1] &quot;chr7:112391225-112392445&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
39 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-9.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-9">
Past versions of unnamed-chunk-4-9.png
</button>
</p>
<div id="fig-unnamed-chunk-4-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-9.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    30    2
  TRUE      4    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 47.08704
sample estimates:
odds ratio 
         0 

[1] &quot;chr8:42391188-42392334&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
34 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-10.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-10">
Past versions of unnamed-chunk-4-10.png
</button>
</p>
<div id="fig-unnamed-chunk-4-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-10.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    13    1
  TRUE     17    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.4516
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 32.11795
sample estimates:
odds ratio 
         0 

[1] &quot;chr8:108248278-108249360&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
33 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-11.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-11">
Past versions of unnamed-chunk-4-11.png
</button>
</p>
<div id="fig-unnamed-chunk-4-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-11.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     8    7
  TRUE      9    7

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1731173 4.5575374
sample estimates:
odds ratio 
 0.8922766 

[1] &quot;chr8:144428066-144429152&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
21 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-12.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-12">
Past versions of unnamed-chunk-4-12.png
</button>
</p>
<div id="fig-unnamed-chunk-4-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-12.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr9:6757448-6758580&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
29 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-13.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-13">
Past versions of unnamed-chunk-4-13.png
</button>
</p>
<div id="fig-unnamed-chunk-4-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-13.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    11    3
  TRUE      7    7

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.2365
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.5563296 28.3544749
sample estimates:
odds ratio 
   3.49268 

[1] &quot;chr9:77176930-77178020&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
26 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-14.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-14">
Past versions of unnamed-chunk-4-14.png
</button>
</p>
<div id="fig-unnamed-chunk-4-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-14.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr9:122159264-122160340&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-15.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-15">
Past versions of unnamed-chunk-4-15.png
</button>
</p>
<div id="fig-unnamed-chunk-4-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-15.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     6   17
  TRUE      1    3

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.06772333 64.70828647
sample estimates:
odds ratio 
  1.056609 

[1] &quot;chr10:104121384-104122530&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
28 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-16.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-16">
Past versions of unnamed-chunk-4-16.png
</button>
</p>
<div id="fig-unnamed-chunk-4-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-16.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    22    1
  TRUE      3    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.0000 297.6142
sample estimates:
odds ratio 
         0 

[1] &quot;chr11:65569529-65570607&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
43 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-17.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-17">
Past versions of unnamed-chunk-4-17.png
</button>
</p>
<div id="fig-unnamed-chunk-4-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-17.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    19    5
  TRUE     15    2

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.6786
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.04325061 3.71114811
sample estimates:
odds ratio 
 0.5147433 

[1] &quot;chr12:120116355-120117431&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
20 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-18.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-18">
Past versions of unnamed-chunk-4-18.png
</button>
</p>
<div id="fig-unnamed-chunk-4-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-18.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    13    1
  TRUE      5    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.5211
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.02764121 219.03192655
sample estimates:
odds ratio 
  2.463069 

[1] &quot;chr12:123533133-123534275&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
34 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-19.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-19">
Past versions of unnamed-chunk-4-19.png
</button>
</p>
<div id="fig-unnamed-chunk-4-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-19.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr13:76991491-76992655&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
33 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-20.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-20">
Past versions of unnamed-chunk-4-20.png
</button>
</p>
<div id="fig-unnamed-chunk-4-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-20.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    21    1
  TRUE      4    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.0000 213.7868
sample estimates:
odds ratio 
         0 

[1] &quot;chr14:22588779-22589885&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
26 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-21.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-21">
Past versions of unnamed-chunk-4-21.png
</button>
</p>
<div id="fig-unnamed-chunk-4-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-21.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     9    3
  TRUE      9    2

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.04602836 7.50950249
sample estimates:
odds ratio 
 0.6784617 

[1] &quot;chr15:63600279-63601501&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
28 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-22.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-22">
Past versions of unnamed-chunk-4-22.png
</button>
</p>
<div id="fig-unnamed-chunk-4-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-22.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    24    2
  TRUE      2    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 89.41917
sample estimates:
odds ratio 
         0 

[1] &quot;chr15:65529892-65530978&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
26 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-23.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-23">
Past versions of unnamed-chunk-4-23.png
</button>
</p>
<div id="fig-unnamed-chunk-4-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-23.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr16:50065406-50066494&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
37 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-24.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-24">
Past versions of unnamed-chunk-4-24.png
</button>
</p>
<div id="fig-unnamed-chunk-4-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-24.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     7    7
  TRUE     10    9

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1830437 4.4247043
sample estimates:
odds ratio 
 0.9028811 

[1] &quot;chr16:67480695-67481811&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
33 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-25.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-25">
Past versions of unnamed-chunk-4-25.png
</button>
</p>
<div id="fig-unnamed-chunk-4-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-25.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    16   10
  TRUE      2    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.01234863 17.44934126
sample estimates:
odds ratio 
 0.8060356 

[1] &quot;chr17:49361764-49362888&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
31 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-26.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-26">
Past versions of unnamed-chunk-4-26.png
</button>
</p>
<div id="fig-unnamed-chunk-4-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-26.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    18    2
  TRUE      9    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 12.06536
sample estimates:
odds ratio 
         0 

[1] &quot;chr19:4400023-4401133&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
26 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-27.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-27">
Past versions of unnamed-chunk-4-27.png
</button>
</p>
<div id="fig-unnamed-chunk-4-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-27.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     9    1
  TRUE     11    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.009573596 71.239880135
sample estimates:
odds ratio 
 0.8257258 

[1] &quot;chr19:10928117-10929247&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
27 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-28.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-28">
Past versions of unnamed-chunk-4-28.png
</button>
</p>
<div id="fig-unnamed-chunk-4-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-28.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    10    6
  TRUE      6    4

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1587713 7.3002652
sample estimates:
odds ratio 
  1.106601 

[1] &quot;chr19:10960352-10961542&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
25 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-29.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-29">
Past versions of unnamed-chunk-4-29.png
</button>
</p>
<div id="fig-unnamed-chunk-4-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-29.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr19:16627850-16628986&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-30.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-30">
Past versions of unnamed-chunk-4-30.png
</button>
</p>
<div id="fig-unnamed-chunk-4-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-30.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    16    6
  TRUE      4    2

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.09563544 12.44587051
sample estimates:
odds ratio 
   1.31925 

[1] &quot;chr19:55599637-55600723&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
24 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-31.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-31">
Past versions of unnamed-chunk-4-31.png
</button>
</p>
<div id="fig-unnamed-chunk-4-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-31.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr19:56120365-56121553&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
32 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-32.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-32">
Past versions of unnamed-chunk-4-32.png
</button>
</p>
<div id="fig-unnamed-chunk-4-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-32.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    21    0
  TRUE      9    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.3226
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.05384599        Inf
sample estimates:
odds ratio 
       Inf 

[1] &quot;chr19:56403514-56404588&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
38 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-33.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-33">
Past versions of unnamed-chunk-4-33.png
</button>
</p>
<div id="fig-unnamed-chunk-4-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-33.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    21    6
  TRUE      3    5

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.07674
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.8045677 46.4324258
sample estimates:
odds ratio 
  5.485051 

[1] &quot;chr20:62795885-62796965&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
31 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-34.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-34">
Past versions of unnamed-chunk-4-34.png
</button>
</p>
<div id="fig-unnamed-chunk-4-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-34.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    15   10
  TRUE      3    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.008676444 7.469107912
sample estimates:
odds ratio 
 0.5111917 

[1] &quot;chr20:64079354-64080472&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
23 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-35.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-35">
Past versions of unnamed-chunk-4-35.png
</button>
</p>
<div id="fig-unnamed-chunk-4-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-35.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;chr21:33324371-33325519&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
35 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-36.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-36">
Past versions of unnamed-chunk-4-36.png
</button>
</p>
<div id="fig-unnamed-chunk-4-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-36.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     8   13
  TRUE      4    7

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1910659 6.6984469
sample estimates:
odds ratio 
  1.074474 

[1] &quot;chr22:24010669-24011749&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
36 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-37.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-37">
Past versions of unnamed-chunk-4-37.png
</button>
</p>
<div id="fig-unnamed-chunk-4-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-37.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    25    3
  TRUE      5    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 15.14882
sample estimates:
odds ratio 
         0 

[1] &quot;chr22:27918755-27919957&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
34 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-38.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-38">
Past versions of unnamed-chunk-4-38.png
</button>
</p>
<div id="fig-unnamed-chunk-4-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/fiberseq/blob/79c7f227b0a194ed564cd7eef54adb4afb5f186e/docs/figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-4-38.png" target="_blank">79c7f22</a>
</td>
<td>
XSun
</td>
<td>
2025-09-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     9    1
  TRUE     10   10

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.04851
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.8725463 431.9797850
sample estimates:
odds ratio 
  8.419868 </code></pre>
</div>
</div>
<div id="motifs-are-500-1000bp-apart" class="section level1">
<h1>Motifs are 500-1000bp apart</h1>
<div id="extract-the-regions-containing-both-tf-motif-sites-1"
class="section level2">
<h2>Extract the regions containing both TF motif sites</h2>
<p>Regions containing both TFs were extracted, and only those in which
the TFs were separated by more than <code>500 bp</code> but less than
<code>1000 bp</code> were retained. Each motif was then extended by 500
bp on both sides to define the final regions.</p>
<pre class="r"><code>motif_sites_motif1 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF1,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif1_chip1 &lt;- motif_sites_motif1[motif_sites_motif1$chip_label == 1,]

motif_sites_motif2 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF2,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif2_chip1 &lt;- motif_sites_motif1[motif_sites_motif2$chip_label == 1,]

motif1 &lt;- motif_sites_motif1_chip1
motif1 &lt;- motif1[!is.na(motif1$start),]
motif2 &lt;- motif_sites_motif2_chip1
motif2 &lt;- motif2[!is.na(motif2$start),]

## Convert to GRanges
gr_motif1 &lt;- GRanges(seqnames = motif1$chr,
                 ranges   = IRanges(start = motif1$start+100, end = motif1$end-100),
                 site     = motif1$name)

gr_motif2 &lt;- GRanges(seqnames = motif2$chr,
                 ranges   = IRanges(start = motif2$start+100, end = motif2$end-100),
                 site     = motif2$name)

## Define distance cutoff
dist_cutoff_large &lt;- 1000
dist_cutoff_small &lt;- 500

## Find nearest motif2 site for each motif1 site
nearest_hits &lt;- distanceToNearest(gr_motif1, gr_motif2)

## Filter by distance threshold
close_pairs &lt;- nearest_hits[mcols(nearest_hits)$distance &lt;= dist_cutoff_large &amp; mcols(nearest_hits)$distance &gt;= dist_cutoff_small]

## Extract regions
motif1_close &lt;- gr_motif1[queryHits(close_pairs)]
motif2_close &lt;- gr_motif2[subjectHits(close_pairs)]

## Make a data.frame
df_close &lt;- data.frame(
  chr = as.character(seqnames(motif1_close)),
  motif1_start = start(motif1_close),
  motif1_end   = end(motif1_close),
  motif1_site  = motif1_close$site,
  motif2_start  = start(motif2_close),
  motif2_end    = end(motif2_close),
  motif2_site   = motif2_close$site,
  distance   = mcols(close_pairs)$distance
)

DT::datatable(df_close,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Motif sites selected&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-9fa6da26214ebfd05e61" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9fa6da26214ebfd05e61">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Motif sites selected<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],["chr1","chr7","chr7","chr7","chr10","chr10","chr10","chr10","chr16","chr16","chr19","chr19","chr19","chr21"],[234610967,74657617,74657622,74657623,68075397,68075398,68075449,68075450,680334,680335,55259110,55259111,55259168,34073182],[234610978,74657628,74657633,74657634,68075408,68075409,68075460,68075461,680345,680346,55259121,55259122,55259179,34073193],["site2743","site12418","site12419","site12420","site16790","site16791","site16792","site16793","site23847","site23848","site29940","site29941","site29942","site31624"],[234610233,74658386,74658386,74658386,68076241,68076241,68076241,68076241,679810,679810,55258588,55258588,55258588,34073776],[234610244,74658397,74658397,74658397,68076252,68076252,68076252,68076252,679821,679821,55258599,55258599,55258599,34073787],["site2741","site12421","site12421","site12421","site16794","site16794","site16794","site16794","site23846","site23846","site29937","site29937","site29937","site31626"],[722,757,752,751,832,831,780,779,512,513,510,511,568,582]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>motif1_start<\/th>\n      <th>motif1_end<\/th>\n      <th>motif1_site<\/th>\n      <th>motif2_start<\/th>\n      <th>motif2_end<\/th>\n      <th>motif2_site<\/th>\n      <th>distance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_summary &lt;- df_close %&gt;%
  group_by(chr, motif2_site, motif2_start, motif2_end) %&gt;%
  summarise(
    min_motif1_start = min(motif1_start),
    max_motif1_end   = max(motif1_end),
    .groups = &quot;drop&quot;
  )

df_final &lt;- df_summary %&gt;%
  transmute(
    chr,
    start = pmin(motif2_start, min_motif1_start) - 500,
    end   = pmax(motif2_end,   max_motif1_end)   + 500
  )

df_final &lt;- df_final %&gt;%
  mutate(chr_num = as.numeric(gsub(&quot;chr&quot;, &quot;&quot;, chr))) %&gt;%
  arrange(chr_num, start) %&gt;%
  select(-chr_num)

DT::datatable(df_final,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Regions selected for plotting &#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-1b1a8ec0941961691c13" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1b1a8ec0941961691c13">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Regions selected for plotting <\/caption>","data":[["1","2","3","4","5","6"],["chr1","chr7","chr10","chr16","chr19","chr21"],[234609733,74657117,68074897,679310,55258088,34072682],[234611478,74658897,68076752,680846,55259679,34074287]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualization-and-co-accessibility-test-1"
class="section level2">
<h2>Visualization and co-accessibility test</h2>
<p>For visualization, highlighted windows are the ones
<code>±7 bp</code> around each motif.</p>
<p>To determine whether a TF has a footprint, footprint data were
extracted from <code>res$fiberHMM_fps_infire</code>.</p>
<p>Footprints were retained if they (1) had a FiberHMM footprint size
&lt;40 bp and (2) overlapped with the highlighted motif windows.</p>
<p>Next, reads covering both motif regions were extracted, and the
presence or absence of footprints for the two TFs was counted.</p>
<p>Last, we did FET</p>
<pre class="r"><code>out_dir=&quot;/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results&quot;
sample_name=&quot;LCL_18508&quot;

df_close_nodup &lt;- df_close[!duplicated(df_close$motif2_site),]

#for (i in 1:3){
for (i in 1:nrow(df_final)){
  
  test_range_str &lt;- paste0(df_final$chr[i],&quot;:&quot;,df_final$start[i],&quot;-&quot;,df_final$end[i])
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)
  
  region &lt;- get_region(result_dir)
  subtitle &lt;- sprintf(&quot;%s:%s-%s&quot;, region$chr, region$start, region$end)
  
  print(subtitle)
  
  df_close_region &lt;- df_close[df_close$motif2_site == df_close_nodup$motif2_site[i],]
  min_start_motif1 &lt;- min(df_close_region$motif1_start)
  max_end_motif1   &lt;- max(df_close_region$motif1_end)
  
  highlight_windows.df &lt;- data.frame(start = c(min_start_motif1 - 7,df_close_nodup$motif2_start[i] -7 ), 
                              end = c(max_end_motif1 + 7 ,df_close_nodup$motif2_end[i] +7))
  
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)

  
  res &lt;- plot_region_reads(result_dir, 
                         filter_fire_reads = FALSE,
                         cover_region_only = FALSE,
                         cluster_reads = TRUE,
                         select_A_only = TRUE,
                         split_haplotype = FALSE,
                         highlight_window = highlight_windows.df,
                         color_highlight_window = &quot;#ade6b3&quot;,
                         plot_name = plot_name,
                         subtitle = subtitle)

  region &lt;- res$region
  window_size &lt;- res$window_size
  rids_df &lt;- res$rids_df
  pileup &lt;- res$pileup
  reads &lt;- res$reads
  fire &lt;- res$fire
  fdr_fire &lt;- res$fdr_fire
  fiberHMM_fps &lt;- res$fiberHMM_fps
  gg &lt;- res$gg
  
  
  print(cowplot::plot_grid(
    gg$pileup,
    gg$dimelo_reads,
    gg$fire_fdr,
    gg$fire_reads,
    gg$fiberHMM,
    ncol = 1,
    rel_heights = c(4,8,2,8,8),
    align = &quot;v&quot;,
    axis = &quot;b&quot;
  ))

  
  fps &lt;- res$fiberHMM_fps_infire
  fps_rid &lt;- fps$RID
  
  #fps_highlight_motif1 &lt;- fps[fps$start &gt; highlight_windows.df$start[1] &amp; fps$end &lt; highlight_windows.df$end[1], ]
  fps_highlight_motif1 &lt;- fps[fps$start &lt;= highlight_windows.df$end[1] &amp; fps$end &gt;= highlight_windows.df$start[1], ]
  fps_tf_motif1 &lt;- fps_highlight_motif1[fps_highlight_motif1$class == &quot;&lt; 40bp&quot;,]
  rid_motif1 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= max_end_motif1 &amp; res$rids_df$end &gt;= min_start_motif1])
  
  
  #fps_highlight_motif2 &lt;- fps[fps$start &gt; highlight_windows.df$start[2] &amp; fps$end &lt; highlight_windows.df$end[2], ]
  fps_highlight_motif2 &lt;- fps[fps$start &lt;= highlight_windows.df$end[2] &amp; fps$end &gt;= highlight_windows.df$start[2], ]
  fps_tf_motif2 &lt;- fps_highlight_motif2[fps_highlight_motif2$class == &quot;&lt; 40bp&quot;,]
  rid_motif2 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= df_close_nodup$motif2_end[i] &amp; res$rids_df$end &gt;= df_close_nodup$motif2_start[i]])
  
  RIDs_overlap_both_regions &lt;- intersect(rid_motif1, rid_motif2)
  
  RIDs_access.df &lt;- data.frame(RID = RIDs_overlap_both_regions, 
                               TF1 = FALSE,
                               TF2 = FALSE)
  
  RIDs_access.df$TF1[RIDs_access.df$RID %in% unique(fps_tf_motif1$RID)] &lt;- TRUE
  RIDs_access.df$TF2[RIDs_access.df$RID %in% unique(fps_tf_motif2$RID)] &lt;- TRUE
  
  
  # print(RIDs_access.df)
  
  if(!(sum(RIDs_access.df$TF1) &gt; 0 &amp; sum(RIDs_access.df$TF2) &gt; 0)) next 
  
  cat(&quot;Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...\n&quot;)
  print(contingency_table &lt;- table(RIDs_access.df[,c(&quot;TF1&quot;, &quot;TF2&quot;)]))
  
  print(fisher_res &lt;- fisher.test(contingency_table))
  
}</code></pre>
<pre><code>[1] &quot;chr1:234609732-234611478&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    22    1
  TRUE      1    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.0000 884.6154
sample estimates:
odds ratio 
         0 

[1] &quot;chr7:74657115-74658897&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
25 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-6-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr10:68074896-68076752&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
24 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-6-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr16:679308-680846&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
36 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-6-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr19:55258087-55259679&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
31 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-6-5.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    23    1
  TRUE      3    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.0000 310.4912
sample estimates:
odds ratio 
         0 

[1] &quot;chr21:34072681-34074287&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
34 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-6-6.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    29    1
  TRUE      1    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
    0.000 1149.013
sample estimates:
odds ratio 
         0 </code></pre>
</div>
</div>
<div id="motifs-are-1000-2000bp-apart" class="section level1">
<h1>Motifs are 1000-2000bp apart</h1>
<div id="extract-the-regions-containing-both-tf-motif-sites-2"
class="section level2">
<h2>Extract the regions containing both TF motif sites</h2>
<p>Regions containing both TFs were extracted, and only those in which
the TFs were separated by more than <code>1000 bp</code> but less than
<code>2000 bp</code> were retained. Each motif was then extended by 500
bp on both sides to define the final regions.</p>
<pre class="r"><code>motif_sites_motif1 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF1,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif1_chip1 &lt;- motif_sites_motif1[motif_sites_motif1$chip_label == 1,]

motif_sites_motif2 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF2,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif2_chip1 &lt;- motif_sites_motif1[motif_sites_motif2$chip_label == 1,]

motif1 &lt;- motif_sites_motif1_chip1
motif1 &lt;- motif1[!is.na(motif1$start),]
motif2 &lt;- motif_sites_motif2_chip1
motif2 &lt;- motif2[!is.na(motif2$start),]

## Convert to GRanges
gr_motif1 &lt;- GRanges(seqnames = motif1$chr,
                 ranges   = IRanges(start = motif1$start+100, end = motif1$end-100),
                 site     = motif1$name)

gr_motif2 &lt;- GRanges(seqnames = motif2$chr,
                 ranges   = IRanges(start = motif2$start+100, end = motif2$end-100),
                 site     = motif2$name)

## Define distance cutoff
dist_cutoff_large &lt;- 2000
dist_cutoff_small &lt;- 1000

## Find nearest motif2 site for each motif1 site
nearest_hits &lt;- distanceToNearest(gr_motif1, gr_motif2)

## Filter by distance threshold
close_pairs &lt;- nearest_hits[mcols(nearest_hits)$distance &lt;= dist_cutoff_large &amp; mcols(nearest_hits)$distance &gt;= dist_cutoff_small]

## Extract regions
motif1_close &lt;- gr_motif1[queryHits(close_pairs)]
motif2_close &lt;- gr_motif2[subjectHits(close_pairs)]

## Make a data.frame
df_close &lt;- data.frame(
  chr = as.character(seqnames(motif1_close)),
  motif1_start = start(motif1_close),
  motif1_end   = end(motif1_close),
  motif1_site  = motif1_close$site,
  motif2_start  = start(motif2_close),
  motif2_end    = end(motif2_close),
  motif2_site   = motif2_close$site,
  distance   = mcols(close_pairs)$distance
)

DT::datatable(df_close,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Motif sites selected&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-d3bdb3a70b69fe9ab76d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d3bdb3a70b69fe9ab76d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Motif sites selected<\/caption>","data":[["1","2","3","4","5","6"],["chr7","chr7","chr19","chr19","chr19","chr19"],[100428316,100428317,3985552,3985597,3985598,42325409],[100428327,100428328,3985563,3985608,3985609,42325420],["site12626","site12627","site28236","site28237","site28238","site29453"],[100429785,100429785,3986925,3986925,3986925,42323950],[100429796,100429796,3986936,3986936,3986936,42323961],["site12630","site12630","site28239","site28239","site28239","site29451"],[1457,1456,1361,1316,1315,1447]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>motif1_start<\/th>\n      <th>motif1_end<\/th>\n      <th>motif1_site<\/th>\n      <th>motif2_start<\/th>\n      <th>motif2_end<\/th>\n      <th>motif2_site<\/th>\n      <th>distance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_summary &lt;- df_close %&gt;%
  group_by(chr, motif2_site, motif2_start, motif2_end) %&gt;%
  summarise(
    min_motif1_start = min(motif1_start),
    max_motif1_end   = max(motif1_end),
    .groups = &quot;drop&quot;
  )

df_final &lt;- df_summary %&gt;%
  transmute(
    chr,
    start = pmin(motif2_start, min_motif1_start) - 500,
    end   = pmax(motif2_end,   max_motif1_end)   + 500
  )

df_final &lt;- df_final %&gt;%
  mutate(chr_num = as.numeric(gsub(&quot;chr&quot;, &quot;&quot;, chr))) %&gt;%
  arrange(chr_num, start) %&gt;%
  select(-chr_num)

DT::datatable(df_final,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Regions selected for plotting &#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-e41253b73217ca07c332" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e41253b73217ca07c332">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Regions selected for plotting <\/caption>","data":[["1","2","3"],["chr7","chr19","chr19"],[100427816,3985052,42323450],[100430296,3987436,42325920]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualization-and-co-accessibility-test-2"
class="section level2">
<h2>Visualization and co-accessibility test</h2>
<p>For visualization, highlighted windows are the ones
<code>±7 bp</code> around each motif.</p>
<p>To determine whether a TF has a footprint, footprint data were
extracted from <code>res$fiberHMM_fps_infire</code>.</p>
<p>Footprints were retained if they (1) had a FiberHMM footprint size
&lt;40 bp and (2) overlapped with the highlighted motif windows.</p>
<p>Next, reads covering both motif regions were extracted, and the
presence or absence of footprints for the two TFs was counted.</p>
<pre class="r"><code>out_dir=&quot;/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results&quot;
sample_name=&quot;LCL_18508&quot;

df_close_nodup &lt;- df_close[!duplicated(df_close$motif2_site),]

#for (i in 1:3){
for (i in 1:nrow(df_final)){
  
  test_range_str &lt;- paste0(df_final$chr[i],&quot;:&quot;,df_final$start[i],&quot;-&quot;,df_final$end[i])
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)
  
  region &lt;- get_region(result_dir)
  subtitle &lt;- sprintf(&quot;%s:%s-%s&quot;, region$chr, region$start, region$end)
  
  print(subtitle)
  
  df_close_region &lt;- df_close[df_close$motif2_site == df_close_nodup$motif2_site[i],]
  min_start_motif1 &lt;- min(df_close_region$motif1_start)
  max_end_motif1   &lt;- max(df_close_region$motif1_end)
  
  highlight_windows.df &lt;- data.frame(start = c(min_start_motif1 - 7,df_close_nodup$motif2_start[i] -7 ), 
                              end = c(max_end_motif1 + 7 ,df_close_nodup$motif2_end[i] +7))
  
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)

  
  res &lt;- plot_region_reads(result_dir, 
                         filter_fire_reads = FALSE,
                         cover_region_only = FALSE,
                         cluster_reads = TRUE,
                         select_A_only = TRUE,
                         split_haplotype = FALSE,
                         highlight_window = highlight_windows.df,
                         color_highlight_window = &quot;#ade6b3&quot;,
                         plot_name = plot_name,
                         subtitle = subtitle)

  region &lt;- res$region
  window_size &lt;- res$window_size
  rids_df &lt;- res$rids_df
  pileup &lt;- res$pileup
  reads &lt;- res$reads
  fire &lt;- res$fire
  fdr_fire &lt;- res$fdr_fire
  fiberHMM_fps &lt;- res$fiberHMM_fps
  gg &lt;- res$gg
  
  
  print(cowplot::plot_grid(
    gg$pileup,
    gg$dimelo_reads,
    gg$fire_fdr,
    gg$fire_reads,
    gg$fiberHMM,
    ncol = 1,
    rel_heights = c(4,8,2,8,8),
    align = &quot;v&quot;,
    axis = &quot;b&quot;
  ))

  
  fps &lt;- res$fiberHMM_fps_infire
  fps_rid &lt;- fps$RID
  
  #fps_highlight_motif1 &lt;- fps[fps$start &gt; highlight_windows.df$start[1] &amp; fps$end &lt; highlight_windows.df$end[1], ]
  fps_highlight_motif1 &lt;- fps[fps$start &lt;= highlight_windows.df$end[1] &amp; fps$end &gt;= highlight_windows.df$start[1], ]
  fps_tf_motif1 &lt;- fps_highlight_motif1[fps_highlight_motif1$class == &quot;&lt; 40bp&quot;,]
  rid_motif1 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= max_end_motif1 &amp; res$rids_df$end &gt;= min_start_motif1])
  
  
  #fps_highlight_motif2 &lt;- fps[fps$start &gt; highlight_windows.df$start[2] &amp; fps$end &lt; highlight_windows.df$end[2], ]
  fps_highlight_motif2 &lt;- fps[fps$start &lt;= highlight_windows.df$end[2] &amp; fps$end &gt;= highlight_windows.df$start[2], ]
  fps_tf_motif2 &lt;- fps_highlight_motif2[fps_highlight_motif2$class == &quot;&lt; 40bp&quot;,]
  rid_motif2 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= df_close_nodup$motif2_end[i] &amp; res$rids_df$end &gt;= df_close_nodup$motif2_start[i]])
  
  RIDs_overlap_both_regions &lt;- intersect(rid_motif1, rid_motif2)
  
  RIDs_access.df &lt;- data.frame(RID = RIDs_overlap_both_regions, 
                               TF1 = FALSE,
                               TF2 = FALSE)
  
  RIDs_access.df$TF1[RIDs_access.df$RID %in% unique(fps_tf_motif1$RID)] &lt;- TRUE
  RIDs_access.df$TF2[RIDs_access.df$RID %in% unique(fps_tf_motif2$RID)] &lt;- TRUE
  
  
  # print(RIDs_access.df)
  
  if(!(sum(RIDs_access.df$TF1) &gt; 0 &amp; sum(RIDs_access.df$TF2) &gt; 0)) next 
  
  cat(&quot;Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...\n&quot;)
  print(contingency_table &lt;- table(RIDs_access.df[,c(&quot;TF1&quot;, &quot;TF2&quot;)]))
  
  print(fisher_res &lt;- fisher.test(contingency_table))
  
}</code></pre>
<pre><code>[1] &quot;chr7:100427814-100430296&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
32 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     7    4
  TRUE     11    4

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.6828
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.08679616 4.74574019
sample estimates:
odds ratio 
 0.6477507 

[1] &quot;chr19:3985050-3987436&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-8-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr19:42323448-42325920&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
35 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-8-3.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="motifs-are-2000-5000bp-apart" class="section level1">
<h1>Motifs are 2000-5000bp apart</h1>
<div id="extract-the-regions-containing-both-tf-motif-sites-3"
class="section level2">
<h2>Extract the regions containing both TF motif sites</h2>
<p>Regions containing both TFs were extracted, and only those in which
the TFs were separated by more than <code>2000 bp</code> but less than
<code>5000 bp</code> were retained. Each motif was then extended by 500
bp on both sides to define the final regions.</p>
<pre class="r"><code>motif_sites_motif1 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF1,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif1_chip1 &lt;- motif_sites_motif1[motif_sites_motif1$chip_label == 1,]

motif_sites_motif2 &lt;- readRDS(paste0(&#39;/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs/&#39;,TF2,&#39;.GM12878.sites.chip.labels.rds&#39;))
motif_sites_motif2_chip1 &lt;- motif_sites_motif1[motif_sites_motif2$chip_label == 1,]

motif1 &lt;- motif_sites_motif1_chip1
motif1 &lt;- motif1[!is.na(motif1$start),]
motif2 &lt;- motif_sites_motif2_chip1
motif2 &lt;- motif2[!is.na(motif2$start),]

## Convert to GRanges
gr_motif1 &lt;- GRanges(seqnames = motif1$chr,
                 ranges   = IRanges(start = motif1$start+100, end = motif1$end-100),
                 site     = motif1$name)

gr_motif2 &lt;- GRanges(seqnames = motif2$chr,
                 ranges   = IRanges(start = motif2$start+100, end = motif2$end-100),
                 site     = motif2$name)

## Define distance cutoff
dist_cutoff_large &lt;- 5000
dist_cutoff_small &lt;- 2000

## Find nearest motif2 site for each motif1 site
nearest_hits &lt;- distanceToNearest(gr_motif1, gr_motif2)

## Filter by distance threshold
close_pairs &lt;- nearest_hits[mcols(nearest_hits)$distance &lt;= dist_cutoff_large &amp; mcols(nearest_hits)$distance &gt;= dist_cutoff_small]

## Extract regions
motif1_close &lt;- gr_motif1[queryHits(close_pairs)]
motif2_close &lt;- gr_motif2[subjectHits(close_pairs)]

## Make a data.frame
df_close &lt;- data.frame(
  chr = as.character(seqnames(motif1_close)),
  motif1_start = start(motif1_close),
  motif1_end   = end(motif1_close),
  motif1_site  = motif1_close$site,
  motif2_start  = start(motif2_close),
  motif2_end    = end(motif2_close),
  motif2_site   = motif2_close$site,
  distance   = mcols(close_pairs)$distance
)

DT::datatable(df_close,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Motif sites selected&#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-6b35d4872c33fe5849ab" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6b35d4872c33fe5849ab">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Motif sites selected<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18"],["chr3","chr4","chr4","chr4","chr4","chr7","chr7","chr7","chr10","chr10","chr11","chr11","chr12","chr12","chr12","chr12","chr15","chr15"],[49786748,26860639,26860640,26860663,26860664,101165574,101165729,101165730,79070011,79070012,64305429,64305430,132710644,132710842,132710843,132710969,42495628,42495629],[49786759,26860650,26860651,26860674,26860675,101165585,101165740,101165741,79070022,79070023,64305440,64305441,132710655,132710853,132710854,132710980,42495639,42495640],["site5802","site7371","site7372","site7373","site7374","site12661","site12662","site12663","site16933","site16934","site18513","site18514","site20911","site20912","site20913","site20914","site23025","site23026"],[49790214,26857702,26857702,26857702,26857702,101162946,101162946,101162946,79067574,79067574,64301111,64301111,132714494,132714494,132714494,132714494,42491125,42491125],[49790225,26857713,26857713,26857713,26857713,101162957,101162957,101162957,79067585,79067585,64301122,64301122,132714505,132714505,132714505,132714505,42491136,42491136],["site5806","site7370","site7370","site7370","site7370","site12657","site12657","site12657","site16932","site16932","site18511","site18511","site20915","site20915","site20915","site20915","site23023","site23023"],[3454,2925,2926,2949,2950,2616,2771,2772,2425,2426,4306,4307,3838,3640,3639,3513,4491,4492]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>motif1_start<\/th>\n      <th>motif1_end<\/th>\n      <th>motif1_site<\/th>\n      <th>motif2_start<\/th>\n      <th>motif2_end<\/th>\n      <th>motif2_site<\/th>\n      <th>distance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3,5,6,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_summary &lt;- df_close %&gt;%
  group_by(chr, motif2_site, motif2_start, motif2_end) %&gt;%
  summarise(
    min_motif1_start = min(motif1_start),
    max_motif1_end   = max(motif1_end),
    .groups = &quot;drop&quot;
  )

df_final &lt;- df_summary %&gt;%
  transmute(
    chr,
    start = pmin(motif2_start, min_motif1_start) - 500,
    end   = pmax(motif2_end,   max_motif1_end)   + 500
  )

df_final &lt;- df_final %&gt;%
  mutate(chr_num = as.numeric(gsub(&quot;chr&quot;, &quot;&quot;, chr))) %&gt;%
  arrange(chr_num, start) %&gt;%
  select(-chr_num)

DT::datatable(df_final,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Regions selected for plotting &#39;),options = list(pageLength = 5) )</code></pre>
<div id="htmlwidget-6e49b5d231d9a4691769" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6e49b5d231d9a4691769">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Regions selected for plotting <\/caption>","data":[["1","2","3","4","5","6","7"],["chr3","chr4","chr7","chr10","chr11","chr12","chr15"],[49786248,26857202,101162446,79067074,64300611,132710144,42490625],[49790725,26861175,101166241,79070523,64305941,132715005,42496140]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="visualization-and-co-accessibility-test-3"
class="section level2">
<h2>Visualization and co-accessibility test</h2>
<p>For visualization, highlighted windows are the ones
<code>±7 bp</code> around each motif.</p>
<p>To determine whether a TF has a footprint, footprint data were
extracted from <code>res$fiberHMM_fps_infire</code>.</p>
<p>Footprints were retained if they (1) had a FiberHMM footprint size
&lt;40 bp and (2) overlapped with the highlighted motif windows.</p>
<p>Next, reads covering both motif regions were extracted, and the
presence or absence of footprints for the two TFs was counted.</p>
<p>Last, we did FET</p>
<pre class="r"><code>out_dir=&quot;/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results&quot;
sample_name=&quot;LCL_18508&quot;

df_close_nodup &lt;- df_close[!duplicated(df_close$motif2_site),]

#for (i in 1:3){
for (i in 1:nrow(df_final)){
  
  test_range_str &lt;- paste0(df_final$chr[i],&quot;:&quot;,df_final$start[i],&quot;-&quot;,df_final$end[i])
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)
  
  region &lt;- get_region(result_dir)
  subtitle &lt;- sprintf(&quot;%s:%s-%s&quot;, region$chr, region$start, region$end)
  
  print(subtitle)
  
  df_close_region &lt;- df_close[df_close$motif2_site == df_close_nodup$motif2_site[i],]
  min_start_motif1 &lt;- min(df_close_region$motif1_start)
  max_end_motif1   &lt;- max(df_close_region$motif1_end)
  
  highlight_windows.df &lt;- data.frame(start = c(min_start_motif1 - 7,df_close_nodup$motif2_start[i] -7 ), 
                              end = c(max_end_motif1 + 7 ,df_close_nodup$motif2_end[i] +7))
  
  plot_name=paste0(TF1,&quot;_&quot;,TF2,&quot;_&quot;,sample_name, &quot;_&quot;, test_range_str)
  result_dir = paste0(out_dir, &quot;/&quot;, plot_name, &quot;/parsed&quot;)

  
  res &lt;- plot_region_reads(result_dir, 
                         filter_fire_reads = FALSE,
                         cover_region_only = FALSE,
                         cluster_reads = TRUE,
                         select_A_only = TRUE,
                         split_haplotype = FALSE,
                         highlight_window = highlight_windows.df,
                         color_highlight_window = &quot;#ade6b3&quot;,
                         plot_name = plot_name,
                         subtitle = subtitle)

  region &lt;- res$region
  window_size &lt;- res$window_size
  rids_df &lt;- res$rids_df
  pileup &lt;- res$pileup
  reads &lt;- res$reads
  fire &lt;- res$fire
  fdr_fire &lt;- res$fdr_fire
  fiberHMM_fps &lt;- res$fiberHMM_fps
  gg &lt;- res$gg
  
  
  print(cowplot::plot_grid(
    gg$pileup,
    gg$dimelo_reads,
    gg$fire_fdr,
    gg$fire_reads,
    gg$fiberHMM,
    ncol = 1,
    rel_heights = c(4,8,2,8,8),
    align = &quot;v&quot;,
    axis = &quot;b&quot;
  ))

  
  fps &lt;- res$fiberHMM_fps_infire
  fps_rid &lt;- fps$RID
  
  #fps_highlight_motif1 &lt;- fps[fps$start &gt; highlight_windows.df$start[1] &amp; fps$end &lt; highlight_windows.df$end[1], ]
  fps_highlight_motif1 &lt;- fps[fps$start &lt;= highlight_windows.df$end[1] &amp; fps$end &gt;= highlight_windows.df$start[1], ]
  fps_tf_motif1 &lt;- fps_highlight_motif1[fps_highlight_motif1$class == &quot;&lt; 40bp&quot;,]
  rid_motif1 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= max_end_motif1 &amp; res$rids_df$end &gt;= min_start_motif1])
  
  
  #fps_highlight_motif2 &lt;- fps[fps$start &gt; highlight_windows.df$start[2] &amp; fps$end &lt; highlight_windows.df$end[2], ]
  fps_highlight_motif2 &lt;- fps[fps$start &lt;= highlight_windows.df$end[2] &amp; fps$end &gt;= highlight_windows.df$start[2], ]
  fps_tf_motif2 &lt;- fps_highlight_motif2[fps_highlight_motif2$class == &quot;&lt; 40bp&quot;,]
  rid_motif2 &lt;-   as.character(res$rids_df$RID[res$rids_df$start &lt;= df_close_nodup$motif2_end[i] &amp; res$rids_df$end &gt;= df_close_nodup$motif2_start[i]])
  
  RIDs_overlap_both_regions &lt;- intersect(rid_motif1, rid_motif2)
  
  RIDs_access.df &lt;- data.frame(RID = RIDs_overlap_both_regions, 
                               TF1 = FALSE,
                               TF2 = FALSE)
  
  RIDs_access.df$TF1[RIDs_access.df$RID %in% unique(fps_tf_motif1$RID)] &lt;- TRUE
  RIDs_access.df$TF2[RIDs_access.df$RID %in% unique(fps_tf_motif2$RID)] &lt;- TRUE
  
  
  # print(RIDs_access.df)
  
  if(!(sum(RIDs_access.df$TF1) &gt; 0 &amp; sum(RIDs_access.df$TF2) &gt; 0)) next 
  
  cat(&quot;Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...\n&quot;)
  print(contingency_table &lt;- table(RIDs_access.df[,c(&quot;TF1&quot;, &quot;TF2&quot;)]))
  
  print(fisher_res &lt;- fisher.test(contingency_table))
  
}</code></pre>
<pre><code>[1] &quot;chr3:49786247-49790725&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
44 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr4:26857201-26861175&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
28 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-10-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    10    0
  TRUE      4    1

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.3333
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.05128188        Inf
sample estimates:
odds ratio 
       Inf 

[1] &quot;chr7:101162445-101166241&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
38 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-10-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr10:79067073-79070523&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-10-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE    18    1
  TRUE      4    0

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.0000 184.7182
sample estimates:
odds ratio 
         0 

[1] &quot;chr11:64300609-64305941&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
30 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-10-5.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr12:132710143-132715005&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
41 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-10-6.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;chr15:42490624-42496140&quot;
Load region from regions.processed.bed 
Load fiberHMM_res from subset_FiberHMM.bed 
39 reads.</code></pre>
<p><img src="figure/TF_co-acc_NRF1_SP1.Rmd/unnamed-chunk-10-7.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>Create a 2x2 table of coaccessibility and run fisher&#39;s exact test...
       TF2
TF1     FALSE TRUE
  FALSE     7   10
  TRUE      1    4

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.613
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
   0.2019471 157.1154336
sample estimates:
odds ratio 
  2.682749 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] forcats_0.5.1        stringr_1.5.1        dplyr_1.1.4         
 [4] purrr_1.0.2          readr_2.1.2          tidyr_1.3.0         
 [7] tibble_3.2.1         tidyverse_1.3.1      ggplot2_3.5.1       
[10] GenomicRanges_1.48.0 GenomeInfoDb_1.39.9  IRanges_2.30.0      
[13] S4Vectors_0.34.0     BiocGenerics_0.42.0  reshape2_1.4.4      
[16] data.table_1.14.2   

loaded via a namespace (and not attached):
 [1] bitops_1.0-7           fs_1.5.2               lubridate_1.8.0       
 [4] httr_1.4.3             rprojroot_2.0.3        tools_4.2.0           
 [7] backports_1.4.1        bslib_0.3.1            utf8_1.2.2            
[10] R6_2.5.1               DT_0.22                DBI_1.2.2             
[13] colorspace_2.0-3       withr_2.5.0            tidyselect_1.2.0      
[16] compiler_4.2.0         git2r_0.30.1           cli_3.6.1             
[19] rvest_1.0.2            xml2_1.3.3             labeling_0.4.2        
[22] sass_0.4.1             scales_1.3.0           digest_0.6.29         
[25] rmarkdown_2.25         XVector_0.36.0         pkgconfig_2.0.3       
[28] htmltools_0.5.2        dbplyr_2.1.1           fastmap_1.1.0         
[31] highr_0.9              htmlwidgets_1.5.4      rlang_1.1.2           
[34] readxl_1.4.0           rstudioapi_0.13        jquerylib_0.1.4       
[37] generics_0.1.2         farver_2.1.0           jsonlite_1.8.0        
[40] crosstalk_1.2.0        RCurl_1.98-1.7         magrittr_2.0.3        
[43] GenomeInfoDbData_1.2.8 Rcpp_1.0.12            munsell_0.5.0         
[46] fansi_1.0.3            lifecycle_1.0.4        stringi_1.7.6         
[49] whisker_0.4            yaml_2.3.5             zlibbioc_1.42.0       
[52] plyr_1.8.7             grid_4.2.0             promises_1.2.0.1      
[55] crayon_1.5.1           haven_2.5.0            cowplot_1.1.1         
[58] hms_1.1.1              knitr_1.39             pillar_1.9.0          
[61] reprex_2.0.1           glue_1.6.2             evaluate_0.15         
[64] modelr_0.1.8           vctrs_0.6.5            tzdb_0.4.0            
[67] httpuv_1.6.5           cellranger_1.1.0       gtable_0.3.0          
[70] assertthat_0.2.1       xfun_0.41              broom_0.8.0           
[73] later_1.3.0            workflowr_1.7.0        ellipsis_0.3.2        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
