---
title: "Annotate footprints by location"
author: "XSun"
date: "2026-02-16"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

```{r}

suppressPackageStartupMessages({
  library(data.table)
  library(ggplot2)
  library(patchwork)
})


```


# Introduction

We check how many footprints are in 

- Promoter
- Pol2
- linker regions
- OCRs called by FIRE using Fiber-seq data

# Footprint and Peak Calling Data

From Kevin 

### 1. FiberHMM Output

FiberHMM results are available at: `/project/spott/1_Shared_projects/LCL_Fiber_seq/FiberHMM/merged/`

### 2. MACS3 Peak Calling on Footprints

Using the FiberHMM-derived footprints, peaks of different footprint size ranges were called with **MACS3**. The results are located at: `/project/spott/1_Shared_projects/LCL_Fiber_seq/FiberHMM/merged/combined/joint_trained_peaks`

Example files:

```
combined_all_chrs_10-30bp_macs3_q0.1_maxgap1_peaks.narrowPeak.gz
combined_all_chrs_10-30bp_macs3_q0.1_maxgap1_peaks.xls
```

- **`.narrowPeak.gz`**
  - UCSC narrowPeak format  
  - 0-based coordinate system  

- **`.xls`**
  - Standard MACS3 peak output table  
  - 1-based coordinate system  

These files can serve as the starting point for downstream footprint-based analyses.


```{r}

res_counts <- readRDS("/project/spott/xsun/fiberseq/2.annotating_footprints/results/count_fp.RDS")

ggplot(res_counts, aes(x = fp_length, y = n_fp)) +
  geom_col() +
  labs(x = "Footprint length bin",
       y = "Number of footprints") +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

```




# Promoter

Promoter regions were defined using the [GENCODE v49 basic gene annotation (GRCh38.p14)](https://www.gencodegenes.org/human/). 
TSSs were extracted from the GTF file `/project/spott/xsun/fiberseq/2.annotating_footprints/data/gencode.v49.basic.annotation.gtf.gz`. 

The genomic annotations were built using Kevin's code: https://github.com/xinhe-lab/mapgen/blob/main/R/annots.R 

The promoter is defined as the resion **2kb** upstream and **500bp** downstream the TSS

**Step-by-step logic**

(1) Load the GTF and restrict chromosomes

- Imports the GTF into a `GRanges` object (`gtf.gr`)
- Drops everything not on autosomes `chr1`–`chr22` using `seqlevels(..., pruning.mode="coarse")`

(2) Keep only protein-coding genes (Optional)

- Filters the full GTF down to entries with `gene_type == "protein_coding"`
- Extracts **gene-level** ranges (`type == "gene"`) into `genes.gr`

(3) Choose one canonical transcript per gene (longest transcript)

- Converts the GTF to a tibble, keeps only `type == "transcript"`
- Groups by `gene_id`
- Computes transcript length as `abs(end - start)`
- Sorts by decreasing length and keeps the top transcript (`slice(1)`)

This produces a table (`canonical.transcripts`) containing **one transcript_id per gene**.

(4) Build canonical transcript annotation tracks

- Keeps only GTF records whose `transcript_id` is in the chosen canonical set:
  - `gtf.canonical.gr`
- Extracts:
  - `exons.gr` where `type == "exon"`
  - `UTRs.gr` where `type == "UTR"`

(5) Infer introns (gene minus exon intervals)

- Computes introns as:  
  `introns.gr <- setdiff(genes.gr, exons.gr)`
- Uses `findOverlaps(genes.gr, introns.gr)` to map each intron interval back to its gene
- Adds a `gene_name` metadata column to introns based on the overlapping gene

(6) Define promoters (TSS - 2 kb, strand-aware)

Promoters are built from **gene coordinates** using strand:

- For `+` strand genes: promoter = `[gene_start - 2000, gene_start]`
- For `-` strand genes: promoter = `[gene_end, gene_end + 2000]`

Then assigns `gene_name` to each promoter interval.

(7) Save outputs

Creates a named list of `GRanges` objects:

- `exons`
- `introns`
- `UTRs`
- `promoters`
- `genes`


<!-- For transcripts annotated on the positive strand, the TSS was defined as the annotated transcript start coordinate; for transcripts on the negative strand, the TSS was defined as the transcript end coordinate. -->

<!-- Promoter regions were constructed by extending each TSS symmetrically upstream(+200bp) and downstream(-1kb).  -->

<!-- To obtain gene-level promoter regions, transcript-level promoters belonging to the same gene (identified by `gene_id`) were grouped and merged using genomic interval reduction (`reduce()` in *GenomicRanges*), producing the union of all promoter intervals associated with each gene. This approach preserves alternative promoter usage while avoiding double counting across transcript isoforms. -->

<!-- Promoters were stratified by gene biotype (`gene_type` in GENCODE), enabling downstream analysis of footprint enrichment across protein-coding genes, long non-coding RNAs, and other gene classes. -->

<!-- All coordinates were based on the GRCh38 reference assembly and exported in BED format for subsequent overlap analyses. -->


## Protein coding genes only

```{r warning=FALSE, message=FALSE, fig.height=4, fig.width=10}

summary_stats_all <- readRDS("/project/spott/xsun/fiberseq/2.annotating_footprints/results/overlap_with_promoter_TSS-2kb+500bp_proteincoding.RDS")

summary_stats_all[, fp_len_hi := as.integer(sub(".*-(\\d+)bp", "\\1", fp_length))]
setorder(summary_stats_all, fp_len_hi)
summary_stats_all[, fp_length := factor(fp_length, levels = unique(fp_length))]

## long format for percent
pct_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("pct_any", "pct_50", "pct_80", "pct_100"),
  variable.name = "threshold",
  value.name = "pct"
)

## Use stable, ASCII keys in the data
pct_long[, threshold := fifelse(threshold=="pct_any","any",
                        fifelse(threshold=="pct_50","50",
                        fifelse(threshold=="pct_80","80","100")))]
pct_long[, threshold := factor(threshold, levels = c("any","50","80","100"))]

p_pct <- ggplot(pct_long, aes(fp_length, pct, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "% footprints overlapping promoters (protein coding genes only)",
       color = NULL) +
  scale_color_discrete(
    breaks = c("any","50","80","100"),
    labels = c("Any overlap", ">=50% covered", ">=80% covered", "100% covered")
    ## If you insist on ≥, try this instead of the ASCII labels above:
    # labels = c("Any overlap", "\u226550% covered", "\u226580% covered", "100% covered")
  ) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

cnt_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("n_any", "n_50", "n_80", "n_100"),
  variable.name = "threshold",
  value.name = "n"
)

cnt_long[, threshold := fifelse(threshold=="n_any","Any overlap",
                                fifelse(threshold=="n_50","≥50% covered",
                                        fifelse(threshold=="n_80","≥80% covered",
                                                "100% covered")))]
cnt_long[, threshold := factor(threshold,
                               levels = c("Any overlap","≥50% covered","≥80% covered","100% covered"))]

p_cnt <- ggplot(cnt_long, aes(fp_length, n, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "Number of footprints overlapping promoters (protein coding genes only)",
       color = NULL) +
  scale_color_discrete(labels = c(
    "Any overlap",
    ">=50% covered",
    ">=80% covered",
    "100% covered"
  )) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

p_pct + p_cnt


```




## All genes

```{r warning=FALSE, message=FALSE, fig.height=4, fig.width=10}

summary_stats_all <- readRDS("/project/spott/xsun/fiberseq/2.annotating_footprints/results/overlap_with_promoter_TSS-2kb+500bp_proteincoding.RDS")

summary_stats_all[, fp_len_hi := as.integer(sub(".*-(\\d+)bp", "\\1", fp_length))]
setorder(summary_stats_all, fp_len_hi)
summary_stats_all[, fp_length := factor(fp_length, levels = unique(fp_length))]

## long format for percent
pct_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("pct_any", "pct_50", "pct_80", "pct_100"),
  variable.name = "threshold",
  value.name = "pct"
)

## Use stable, ASCII keys in the data
pct_long[, threshold := fifelse(threshold=="pct_any","any",
                        fifelse(threshold=="pct_50","50",
                        fifelse(threshold=="pct_80","80","100")))]
pct_long[, threshold := factor(threshold, levels = c("any","50","80","100"))]

p_pct <- ggplot(pct_long, aes(fp_length, pct, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "% footprints overlapping promoters (all genes)",
       color = NULL) +
  scale_color_discrete(
    breaks = c("any","50","80","100"),
    labels = c("Any overlap", ">=50% covered", ">=80% covered", "100% covered")
    ## If you insist on ≥, try this instead of the ASCII labels above:
    # labels = c("Any overlap", "\u226550% covered", "\u226580% covered", "100% covered")
  ) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

cnt_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("n_any", "n_50", "n_80", "n_100"),
  variable.name = "threshold",
  value.name = "n"
)

cnt_long[, threshold := fifelse(threshold=="n_any","Any overlap",
                                fifelse(threshold=="n_50","≥50% covered",
                                        fifelse(threshold=="n_80","≥80% covered",
                                                "100% covered")))]
cnt_long[, threshold := factor(threshold,
                               levels = c("Any overlap","≥50% covered","≥80% covered","100% covered"))]

p_cnt <- ggplot(cnt_long, aes(fp_length, n, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "Number of footprints overlapping promoters (all genes)",
       color = NULL) +
  scale_color_discrete(labels = c(
    "Any overlap",
    ">=50% covered",
    ">=80% covered",
    "100% covered"
  )) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

p_pct + p_cnt


```




# OCR called by FIRE using Fiber-seq data

`/project/spott/kevinluo/Fiber_seq/results/QTL/fireQTL/consensus_peaks/merged_fire_peaks_31samples.bed.gz`


```{r warning=FALSE, message=FALSE, fig.height=4, fig.width=10}

summary_stats_all <- readRDS("/project/spott/xsun/fiberseq/2.annotating_footprints/results/overlap_with_OCR.RDS")

summary_stats_all[, fp_len_hi := as.integer(sub(".*-(\\d+)bp", "\\1", fp_length))]
setorder(summary_stats_all, fp_len_hi)
summary_stats_all[, fp_length := factor(fp_length, levels = unique(fp_length))]

## long format for percent
pct_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("pct_any", "pct_50", "pct_80", "pct_100"),
  variable.name = "threshold",
  value.name = "pct"
)

## Use stable, ASCII keys in the data
pct_long[, threshold := fifelse(threshold=="pct_any","any",
                        fifelse(threshold=="pct_50","50",
                        fifelse(threshold=="pct_80","80","100")))]
pct_long[, threshold := factor(threshold, levels = c("any","50","80","100"))]

p_pct <- ggplot(pct_long, aes(fp_length, pct, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "% footprints overlapping OCR",
       color = NULL) +
  scale_color_discrete(
    breaks = c("any","50","80","100"),
    labels = c("Any overlap", ">=50% covered", ">=80% covered", "100% covered")
    ## If you insist on ≥, try this instead of the ASCII labels above:
    # labels = c("Any overlap", "\u226550% covered", "\u226580% covered", "100% covered")
  ) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

cnt_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("n_any", "n_50", "n_80", "n_100"),
  variable.name = "threshold",
  value.name = "n"
)

cnt_long[, threshold := fifelse(threshold=="n_any","Any overlap",
                                fifelse(threshold=="n_50","≥50% covered",
                                        fifelse(threshold=="n_80","≥80% covered",
                                                "100% covered")))]
cnt_long[, threshold := factor(threshold,
                               levels = c("Any overlap","≥50% covered","≥80% covered","100% covered"))]

p_cnt <- ggplot(cnt_long, aes(fp_length, n, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "Number of footprints overlapping OCR",
       color = NULL) +
  scale_color_discrete(labels = c(
    "Any overlap",
    ">=50% covered",
    ">=80% covered",
    "100% covered"
  )) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

p_pct + p_cnt


```

# Pol2 

GM12878 downloaded by kevin

`/project/spott/kevinluo/Fiber_seq/data/ChIPseq/GM12878/POLR2A.GM12878.ChIPseq.peaks.ENCFF886PSD.bed.gz`


Others to download: https://www.encodeproject.org/search/?type=Experiment&control_type!=*&status=released&p[…]fic_name=Homo+sapiens&biosample_ontology.term_name=GM12891


```{r warning=FALSE, message=FALSE, fig.height=4, fig.width=10}

summary_stats_all <- readRDS("/project/spott/xsun/fiberseq/2.annotating_footprints/results/overlap_with_pol2.RDS")

summary_stats_all[, fp_len_hi := as.integer(sub(".*-(\\d+)bp", "\\1", fp_length))]
setorder(summary_stats_all, fp_len_hi)
summary_stats_all[, fp_length := factor(fp_length, levels = unique(fp_length))]

## long format for percent
pct_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("pct_any", "pct_50", "pct_80", "pct_100"),
  variable.name = "threshold",
  value.name = "pct"
)

## Use stable, ASCII keys in the data
pct_long[, threshold := fifelse(threshold=="pct_any","any",
                        fifelse(threshold=="pct_50","50",
                        fifelse(threshold=="pct_80","80","100")))]
pct_long[, threshold := factor(threshold, levels = c("any","50","80","100"))]

p_pct <- ggplot(pct_long, aes(fp_length, pct, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "% footprints overlapping Pol2",
       color = NULL) +
  scale_color_discrete(
    breaks = c("any","50","80","100"),
    labels = c("Any overlap", ">=50% covered", ">=80% covered", "100% covered")
    ## If you insist on ≥, try this instead of the ASCII labels above:
    # labels = c("Any overlap", "\u226550% covered", "\u226580% covered", "100% covered")
  ) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

cnt_long <- melt(
  summary_stats_all,
  id.vars = c("fp_length", "fp_len_hi"),
  measure.vars = c("n_any", "n_50", "n_80", "n_100"),
  variable.name = "threshold",
  value.name = "n"
)

cnt_long[, threshold := fifelse(threshold=="n_any","Any overlap",
                                fifelse(threshold=="n_50","≥50% covered",
                                        fifelse(threshold=="n_80","≥80% covered",
                                                "100% covered")))]
cnt_long[, threshold := factor(threshold,
                               levels = c("Any overlap","≥50% covered","≥80% covered","100% covered"))]

p_cnt <- ggplot(cnt_long, aes(fp_length, n, color = threshold, group = threshold)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2.5) +
  labs(x = "Footprint length bin",
       y = "Number of footprints overlapping Pol2",
       color = NULL) +
  scale_color_discrete(labels = c(
    "Any overlap",
    ">=50% covered",
    ">=80% covered",
    "100% covered"
  )) +
  theme_classic(base_size = 13) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))

p_pct + p_cnt

```

