---
title: "Checking how CpG methylation affects TF binding"
author: "XSun"
date: "2025-09-18"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---
```{r message=FALSE,warning=FALSE, fig.height=10, fig.width=10}

library(knitr)
library(data.table)
library(reshape2)
library(GenomicRanges)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(Biostrings)
library(gridExtra)
library(purrr)

source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/process_fiberseq_data.R")
source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/plots.R")

DT::datatable(matrix())
```

# Introduction

We check how CpG methylation affects TF binding here. 

We focus on the regions that only contains one NRF1 motif site.

Details for NRF1 motif: https://jaspar.elixir.no/matrix/MA0506.1/

We just take the consensus into consideration, there are 2 CpG sites that may be methylated: position 2 and position 8. On the opposite strand, they are position 3 and position 9. 

<figure class="half">
    <img src="https://github.com/xsun1229/fiberseq/raw/master/output/MA0506.1.png" width="100%">
</figure>


# Examples

We illustrate the workflow using several examples.

```{r message=FALSE,warning=FALSE, fig.height=10, fig.width=10}

TF1 <- "NRF1"

# ---- Load motif sites ----
motif_sites <- readRDS(file.path(
  "/project/spott/kevinluo/Fiber_seq/processed_data/hg38/TFs",
  paste0(TF1, ".GM12878.sites.chip.labels.rds"))
) %>% filter(chip_label == 1, !is.na(start))

motif_sites <- motif_sites[motif_sites$strand == "+",]

region_df <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_regions.RDS")

out_dir="/project/spott/xsun/fiberseq/1.TFs/plot_region/results/NRF1_allmotifsite_regions/"
sample_name="LCL_18508"

rid_motifsite_mat           <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/rid_motifsite_mat_sites_select.RDS")
rid_motifsite_combined      <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/rid_motifsite_combined_sites_select.RDS")
rid_motifsite_methy         <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/rid_motifsite_methy_sites_select.RDS")
sites_selected              <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/sites_select.RDS")
RIDs_access.df_all          <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/RIDs_access.df_sites_select.RDS")
cpg_status_all              <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/cpg_status_all_sites_select.RDS")
contingency_table_pos1_all   <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/contingency_table_pos1.RDS")
contingency_table_pos2_all   <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/contingency_table_pos2.RDS")
prop_col_pos1_all            <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/contingency_prop_col_pos1.RDS")
prop_col_pos2_all            <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/contingency_prop_col_pos2.RDS")
```



## site1968


```{r results='asis',message=FALSE,warning=FALSE, fig.height=10, fig.width=10}

site <- "site1968"

region_for_site <- region_df %>% filter(grepl(paste0("\\b", site, "\\b"), motifs))

test_range_str <- paste0(region_for_site$chr,":",region_for_site$start,"-",region_for_site$end)
plot_name <- paste0(TF1,"_pairs_",sample_name,"_",test_range_str)
result_dir <- paste0(out_dir, "/", plot_name, "/parsed")

region <- get_region(result_dir)
subtitle <- sprintf("%s:%s-%s", region$chr, region$start, region$end)

# motif coordinates
start <- motif_sites$start[motif_sites$name == site] + 100 
end <- motif_sites$end[motif_sites$name == site] - 100 

region_plot <- list(chr = region$chr,
                      start = start - 150,
                      end = end + 150)

res <- try(get_region_result(result_dir,
                             cover_region_only = FALSE,
                             select_A_only = FALSE,
                             split_haplotype = FALSE,
                             fiberHMM_size_breaks = c(60,100),
                             fiberHMM_cols = c("magenta","royalblue","darkgreen"),
                             highlight_window = data.frame(start=start, end=end),
                             color_highlight_window = "#44d6d8ff",
                             plot_name = plot_name,
                             # cluster = "fiberHMM_configs",
                             # cluster_regions = cluster_regions,
                             show_RIDs = F,
                             region = region_plot,
                             subtitle = subtitle,
                             verbose = T)
)

gg <- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = "v",
  axis = "b"
))


cat("We identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. \n")

cat("We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome")

DT::datatable(rid_motifsite_mat[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('CpG positions in the motif for each read --',site)),options = list(pageLength = 10) )

cat("We then retrieve methylation information from res$read. \n")


DT::datatable(rid_motifsite_methy[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation quality score for each read --',site)),options = list(pageLength = 10) )

cat("By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mC or -mC(methylation on reverse strand)\n")

DT::datatable(rid_motifsite_combined[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation annotation for each read --',site)),options = list(pageLength = 10) )

cat("We summarize the methylation and TF footprint status for each read\n")

DT::datatable(RIDs_access.df_all[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation and TF footprint status for each read --',site)),options = list(pageLength = 10) )

cat("For position 1, we have\n")


kable(as.matrix(contingency_table_pos1_all[[site]]),
      caption = paste("Contingency table for", site))

cat("The proportion of TF footprint in different methylation groups")

kable(as.matrix(prop_col_pos1_all[[site]]),
      caption = paste("Proportion table for", site))

cat("For position 2, we have\n")

kable(as.matrix(contingency_table_pos2_all[[site]]),
      caption = paste("Contingency table for", site))

cat("The proportion of TF footprint in different methylation groups")

kable(as.matrix(prop_col_pos2_all[[site]]),
      caption = paste("Proportion table for", site))



```




## site17866


```{r results='asis',message=FALSE,warning=FALSE, fig.height=10, fig.width=10}

site <- "site17866"

region_for_site <- region_df %>% filter(grepl(paste0("\\b", site, "\\b"), motifs))

test_range_str <- paste0(region_for_site$chr,":",region_for_site$start,"-",region_for_site$end)
plot_name <- paste0(TF1,"_pairs_",sample_name,"_",test_range_str)
result_dir <- paste0(out_dir, "/", plot_name, "/parsed")

region <- get_region(result_dir)
subtitle <- sprintf("%s:%s-%s", region$chr, region$start, region$end)

# motif coordinates
start <- motif_sites$start[motif_sites$name == site] + 100 
end <- motif_sites$end[motif_sites$name == site] - 100 

region_plot <- list(chr = region$chr,
                      start = start - 150,
                      end = end + 150)

res <- try(get_region_result(result_dir,
                             cover_region_only = FALSE,
                             select_A_only = FALSE,
                             split_haplotype = FALSE,
                             fiberHMM_size_breaks = c(60,100),
                             fiberHMM_cols = c("magenta","royalblue","darkgreen"),
                             highlight_window = data.frame(start=start, end=end),
                             color_highlight_window = "#44d6d8ff",
                             plot_name = plot_name,
                             # cluster = "fiberHMM_configs",
                             # cluster_regions = cluster_regions,
                             show_RIDs = F,
                             region = region_plot,
                             subtitle = subtitle,
                             verbose = T)
)

gg <- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = "v",
  axis = "b"
))


cat("We identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. \n")

cat("We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome")

DT::datatable(rid_motifsite_mat[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('CpG positions in the motif for each read --',site)),options = list(pageLength = 10) )

cat("We then retrieve methylation information from res$read. \n")


DT::datatable(rid_motifsite_methy[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation quality score for each read --',site)),options = list(pageLength = 10) )

cat("By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mC or -mC(methylation on reverse strand)\n")

DT::datatable(rid_motifsite_combined[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation annotation for each read --',site)),options = list(pageLength = 10) )

cat("We summarize the methylation and TF footprint status for each read\n")

DT::datatable(RIDs_access.df_all[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation and TF footprint status for each read --',site)),options = list(pageLength = 10) )

cat("For position 1, we have\n")


kable(as.matrix(contingency_table_pos1_all[[site]]),
      caption = paste("Contingency table for", site))

cat("The proportion of TF footprint in different methylation groups")

kable(as.matrix(prop_col_pos1_all[[site]]),
      caption = paste("Proportion table for", site))

cat("For position 2, we have\n")

kable(as.matrix(contingency_table_pos2_all[[site]]),
      caption = paste("Contingency table for", site))

cat("The proportion of TF footprint in different methylation groups")

kable(as.matrix(prop_col_pos2_all[[site]]),
      caption = paste("Proportion table for", site))



```





## site32165


```{r results='asis',message=FALSE,warning=FALSE, fig.height=10, fig.width=10}

site <- "site32165"

region_for_site <- region_df %>% filter(grepl(paste0("\\b", site, "\\b"), motifs))

test_range_str <- paste0(region_for_site$chr,":",region_for_site$start,"-",region_for_site$end)
plot_name <- paste0(TF1,"_pairs_",sample_name,"_",test_range_str)
result_dir <- paste0(out_dir, "/", plot_name, "/parsed")

region <- get_region(result_dir)
subtitle <- sprintf("%s:%s-%s", region$chr, region$start, region$end)

# motif coordinates
start <- motif_sites$start[motif_sites$name == site] + 100 
end <- motif_sites$end[motif_sites$name == site] - 100 

region_plot <- list(chr = region$chr,
                      start = start - 150,
                      end = end + 150)

res <- try(get_region_result(result_dir,
                             cover_region_only = FALSE,
                             select_A_only = FALSE,
                             split_haplotype = FALSE,
                             fiberHMM_size_breaks = c(60,100),
                             fiberHMM_cols = c("magenta","royalblue","darkgreen"),
                             highlight_window = data.frame(start=start, end=end),
                             color_highlight_window = "#44d6d8ff",
                             plot_name = plot_name,
                             # cluster = "fiberHMM_configs",
                             # cluster_regions = cluster_regions,
                             show_RIDs = F,
                             region = region_plot,
                             subtitle = subtitle,
                             verbose = T)
)

gg <- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = "v",
  axis = "b"
))


cat("We identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. \n")

cat("We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome")

DT::datatable(rid_motifsite_mat[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('CpG positions in the motif for each read --',site)),options = list(pageLength = 10) )

cat("We then retrieve methylation information from res$read. \n")


DT::datatable(rid_motifsite_methy[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation quality score for each read --',site)),options = list(pageLength = 10) )

cat("By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mC or -mC(methylation on reverse strand)\n")

DT::datatable(rid_motifsite_combined[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation annotation for each read --',site)),options = list(pageLength = 10) )

cat("We summarize the methylation and TF footprint status for each read\n")

DT::datatable(RIDs_access.df_all[[site]],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Methylation and TF footprint status for each read --',site)),options = list(pageLength = 10) )

cat("For position 1, we have\n")


kable(as.matrix(contingency_table_pos1_all[[site]]),
      caption = paste("Contingency table for", site))

cat("The proportion of TF footprint in different methylation groups")

kable(as.matrix(prop_col_pos1_all[[site]]),
      caption = paste("Proportion table for", site))

cat("For position 2, we have\n")

kable(as.matrix(contingency_table_pos2_all[[site]]),
      caption = paste("Contingency table for", site))

cat("The proportion of TF footprint in different methylation groups")

kable(as.matrix(prop_col_pos2_all[[site]]),
      caption = paste("Proportion table for", site))



```



# Summary for each methylation group

We only consider the motifs with at least one CpGm read. 

We did paired t-test to test if the footprint detected proportion are different. 



```{r message=FALSE,warning=FALSE, fig.height=4, fig.width=10}
contingency_prop_col <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/contingency_prop_col_pos1_sites_select.RDS")

# extract the 2nd row of each table if it's valid
mat <- contingency_prop_col %>%
  imap_dfr(~ {
    if (all(is.nan(.x[,2]))) return(NULL)
    tibble(
      methylated     = .x[2, 1],
      not_methylated = .x[2, 2],
      motif          = .y
    )
  })

# run paired t-test
# t <- t.test(mat$methylated, mat$not_methylated, paired = TRUE)
# pval_label <- paste0("paired t-test p = \n", signif(t$p.value, 3))

# long format for plotting
df_long <- mat %>%
  pivot_longer(cols = c(methylated, not_methylated),
               names_to = "CpG", 
               values_to = "% of Footprint")

# plot
p1 <- ggplot(df_long, aes(x = CpG, y = `% of Footprint`, fill = CpG)) +
  geom_violin(trim = FALSE, alpha = 0.3) +
  geom_jitter(width = 0.15, alpha = 0.6, size = 1.5) +
  theme_minimal() +
  ggtitle("position 1")

contingency_prop_col <- readRDS("/project/spott/xsun/fiberseq/1.TFs/results/NRF1_cpg_1278/contingency_prop_col_pos2_sites_select.RDS")

# extract the 2nd row of each table if it's valid
mat <- contingency_prop_col %>%
  imap_dfr(~ {
    if (all(is.nan(.x[,2]))) return(NULL)
    tibble(
      methylated     = .x[2, 1],
      not_methylated = .x[2, 2],
      motif          = .y
    )
  })

# run paired t-test
# t <- t.test(mat$methylated, mat$not_methylated, paired = TRUE)
# pval_label <- paste0("paired t-test p = \n", signif(t$p.value, 3))

# long format for plotting
df_long <- mat %>%
  pivot_longer(cols = c(methylated, not_methylated),
               names_to = "CpG", 
               values_to = "% of Footprint")

# plot
p2 <- ggplot(df_long, aes(x = CpG, y = `% of Footprint`, fill = CpG)) +
  geom_violin(trim = FALSE, alpha = 0.3) +
  geom_jitter(width = 0.15, alpha = 0.6, size = 1.5) +
  theme_minimal() +
  ggtitle("position 2")


grid.arrange(p1, p2, ncol = 2, top = "Distribution of % of Footprint by CpG")

```