---
title: "Checking how CpG methylation affects TF binding"
author: "XSun"
date: "2025-09-09"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---


We check how CpG methylation affects TF binding here. 

We focus on the regions selected here: https://xsun1229.github.io/fiberseq/TF_co-acc_NRF1_pairs.html

# Examples

We first illustrate the workflow using 2 examples.

## chr10:122879000-122880300

The first example is region chr10:122879500–122879700, focusing on the motif chr10:122879629–122879640.

```{r message=FALSE,warning=FALSE, fig.height=10, fig.width=10}
library(data.table)
library(reshape2)
library(GenomicRanges)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(Biostrings)

source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/process_fiberseq_data.R")
source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/plots.R")

test_range_str <- "chr10:122879000-122880300"
plot_name <- "NRF1_pairs_LCL_18508_chr10:122879000-122880300"
result_dir <- "/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_pairs//NRF1_pairs_LCL_18508_chr10:122879000-122880300/parsed"

region <- get_region(result_dir)
subtitle <- sprintf("%s:%s-%s", region$chr, region$start, region$end)

# motif coordinates
start_motif1 <- 122879629
end_motif1   <- 122879640

region_plot <- list(chr ="chr10",
                    start = 122879500,
                    end = 122879800)

res <- try(assemble_region_result(result_dir,
                       filter_fire_reads = FALSE,
                       cover_region_only = FALSE,
                       select_A_only = FALSE,
                       split_haplotype = FALSE,
                       highlight_window = data.frame(start=c(start_motif1), end=c(end_motif1)),
                       color_highlight_window = "#44d6d8ff",
                       plot_name = plot_name,region = region_plot,
                       subtitle = subtitle)
  )

gg <- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = "v",
  axis = "b"
))


```

Next, we identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. 

```{r}
seq_region <- readDNAStringSet(file.path(result_dir,"/region.fa"))
dna_seq <- seq_region[[1]]
cpg_rel_pos <- start(matchPattern("CG", dna_seq))
header <- names(seq_region)       # e.g. "chr10:122879000-122880300"
parts <- strsplit(header, "[:-]")[[1]]
chrom <- parts[1]
region_start <- as.integer(parts[2])
cpg_df <- data.frame(
  chrom = chrom,
  position = region_start + cpg_rel_pos - 1
)
cpg_df_inmotif <- cpg_df[cpg_df$position >= start_motif1 & cpg_df$position <= end_motif1,]

rid_motif1 <- as.character(res$rids_df$RID[res$rids_df$start <= end_motif1 & res$rids_df$end >= start_motif1])

rid_motifsite_mat <- matrix(
  NA, 
  nrow = length(rid_motif1), 
  ncol = length(start_motif1:end_motif1),
  dimnames = list(rid_motif1, as.character(start_motif1:end_motif1))
)
rid_motifsite_methy <- rid_motifsite_mat

# Mark CpG positions
rid_motifsite_mat[,colnames(rid_motifsite_mat) %in% as.character(cpg_df_inmotif$position)] <- "CpG"

DT::datatable(rid_motifsite_mat,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','CpG positions in the motif for each read'),options = list(pageLength = 10) )


```


We then retrieve methylation information from res$read.

```{r}

# Fill methylation
reads_cpg <- res$reads[res$reads$base == "CG",]
reads_cpg$RID <- as.character(reads_cpg$RID)

idx <- which(reads_cpg$RID %in% rownames(rid_motifsite_methy) &
               reads_cpg$ref_position %in% colnames(rid_motifsite_methy))

for (j in idx) {
  rid_motifsite_methy[reads_cpg$RID[j], as.character(reads_cpg$ref_position[j])] <- reads_cpg$mod_qual[j]
}

DT::datatable(rid_motifsite_methy,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation quality score for each read'),options = list(pageLength = 10) )

stopifnot(identical(dimnames(rid_motifsite_methy), dimnames(rid_motifsite_mat)))

```

By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mCpG. A read is classified as mCpG if it contains at least one methylated CpG.

```{r}
# Combine CpG & methylation
rid_motifsite_combined <- ifelse(
  !is.na(rid_motifsite_methy) & !is.na(rid_motifsite_mat), 
  "mCpG",
  ifelse(!is.na(rid_motifsite_mat), "CpG", NA)
)

DT::datatable(rid_motifsite_combined,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation annotation for each read'),options = list(pageLength = 10) )

```

If a read has at least one mCpG, we annotate this read as mCpG

```{r}
RID_cpg <- data.frame(
  RID = rownames(rid_motifsite_combined),
  status = apply(rid_motifsite_combined, 1, function(x) {
    if ("mCpG" %in% x) "mCpG"
    else if (all(na.omit(x) == "CpG")) "CpG"
    else NA
  }),
  row.names = NULL
)

DT::datatable(RID_cpg,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation info for each read'),options = list(pageLength = 10) )
```

This results in two groups of reads: those containing mCpG and those without methylation. We then compare footprint frequencies between the two groups.

We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome.

```{r}
fps <- res$fiberHMM_fps_infire
fps_TF <- fps[fps$class %in% c("< 40bp", "40-60bp"),]
fps_gr <- GRanges(fps_TF$chr, IRanges(fps_TF$start, fps_TF$end), strand=fps_TF$strand)
mcols(fps_gr) <- fps_TF[, !(names(fps_TF) %in% c("chr","start","end","strand"))]

hw <- GRanges(region$chr, IRanges(start_motif1, end_motif1))
fps_gr <- fps_gr[width(pintersect(fps_gr, hw)) > 0]
rid_motif1_fp <- as.character(fps_gr$RID)

fps <- res$fiberHMM_fps
fps_nucl <- fps[fps$size %in% c("> 100bp", "60-100bp"),]
fps_gr_nucl <- GRanges(fps_nucl$chr, IRanges(fps_nucl$fp_blockStarts, fps_nucl$fp_blockEnds))
mcols(fps_gr_nucl) <- fps_nucl[, !(names(fps_nucl) %in% c("chr","start","end"))]
fps_gr_nucl <- fps_gr_nucl[width(pintersect(fps_gr_nucl, hw, ignore.strand = T)) > 0]
rid_fps_nucl <- as.character(fps_gr_nucl$RID)

# filter rid sets
rid_motif1_without_nucl <- setdiff(rid_motif1, rid_fps_nucl)
RIDs_overlap_both <- intersect(rid_motif1_without_nucl, RID_cpg$RID)

RIDs_access.df <- data.frame(
  RID = RIDs_overlap_both,
  TF = RIDs_overlap_both %in% rid_motif1_fp,
  mCpG = RIDs_overlap_both %in% RID_cpg$RID[RID_cpg$status=="mCpG"]
)

RIDs_access.df$TF <- factor(RIDs_access.df$TF, levels=c(FALSE, TRUE))
RIDs_access.df$mCpG <- factor(RIDs_access.df$mCpG, levels=c(FALSE, TRUE))

print(RIDs_access.df)

contingency_table <- table(RIDs_access.df[,c("TF","mCpG")]) 
prop_col <- round(prop.table(contingency_table, margin = 2) * 100,2)

print(contingency_table)
print(prop_col)

```









## chr7:101217498-101218700

The second example is region chr7:101217498-101218700, focusing on the motif chr7:101218054-101218265.

```{r message=FALSE,warning=FALSE, fig.height=10, fig.width=10}
library(data.table)
library(reshape2)
library(GenomicRanges)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(Biostrings)

source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/process_fiberseq_data.R")
source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/plots.R")

test_range_str <- "chr7:101217500-101218700"
plot_name <- "NRF1_pairs_LCL_18508_chr7:101217500-101218700"
result_dir <- "/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_pairs//NRF1_pairs_LCL_18508_chr7:101217500-101218700/parsed"

region <- get_region(result_dir)
subtitle <- sprintf("%s:%s-%s", region$chr, region$start, region$end)

# motif coordinates
start_motif1 <- 101218154
end_motif1   <- 101218165

region_plot <- list(chr ="chr10",
                    start = 101218000,
                    end = 101218300)

res <- try(assemble_region_result(result_dir,
                       filter_fire_reads = FALSE,
                       cover_region_only = FALSE,
                       select_A_only = FALSE,
                       split_haplotype = FALSE,
                       highlight_window = data.frame(start=c(start_motif1), end=c(end_motif1)),
                       color_highlight_window = "#44d6d8ff",
                       plot_name = plot_name,region = region_plot,
                       subtitle = subtitle)
  )

gg <- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = "v",
  axis = "b"
))


```

Next, we identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. 

```{r}
seq_region <- readDNAStringSet(file.path(result_dir,"/region.fa"))
dna_seq <- seq_region[[1]]
cpg_rel_pos <- start(matchPattern("CG", dna_seq))
header <- names(seq_region)       
parts <- strsplit(header, "[:-]")[[1]]
chrom <- parts[1]
region_start <- as.integer(parts[2])
cpg_df <- data.frame(
  chrom = chrom,
  position = region_start + cpg_rel_pos - 1
)
cpg_df_inmotif <- cpg_df[cpg_df$position >= start_motif1 & cpg_df$position <= end_motif1,]

rid_motif1 <- as.character(res$rids_df$RID[res$rids_df$start <= end_motif1 & res$rids_df$end >= start_motif1])

rid_motifsite_mat <- matrix(
  NA, 
  nrow = length(rid_motif1), 
  ncol = length(start_motif1:end_motif1),
  dimnames = list(rid_motif1, as.character(start_motif1:end_motif1))
)
rid_motifsite_methy <- rid_motifsite_mat

# Mark CpG positions
rid_motifsite_mat[,colnames(rid_motifsite_mat) %in% as.character(cpg_df_inmotif$position)] <- "CpG"

DT::datatable(rid_motifsite_mat,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','CpG positions in the motif for each read'),options = list(pageLength = 10) )


```


We then retrieve methylation information from res$read.

```{r}

# Fill methylation
reads_cpg <- res$reads[res$reads$base == "CG",]
reads_cpg$RID <- as.character(reads_cpg$RID)

idx <- which(reads_cpg$RID %in% rownames(rid_motifsite_methy) &
               reads_cpg$ref_position %in% colnames(rid_motifsite_methy))

for (j in idx) {
  rid_motifsite_methy[reads_cpg$RID[j], as.character(reads_cpg$ref_position[j])] <- reads_cpg$mod_qual[j]
}

DT::datatable(rid_motifsite_methy,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation quality score for each read'),options = list(pageLength = 10) )

stopifnot(identical(dimnames(rid_motifsite_methy), dimnames(rid_motifsite_mat)))

```

By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mCpG. A read is classified as mCpG if it contains at least one methylated CpG.

```{r}
# Combine CpG & methylation
rid_motifsite_combined <- ifelse(
  !is.na(rid_motifsite_methy) & !is.na(rid_motifsite_mat), 
  "mCpG",
  ifelse(!is.na(rid_motifsite_mat), "CpG", NA)
)

DT::datatable(rid_motifsite_combined,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation annotation for each read'),options = list(pageLength = 10) )

```

If a read has at least one mCpG, we annotate this read as mCpG

```{r}
RID_cpg <- data.frame(
  RID = rownames(rid_motifsite_combined),
  status = apply(rid_motifsite_combined, 1, function(x) {
    if ("mCpG" %in% x) "mCpG"
    else if (all(na.omit(x) == "CpG")) "CpG"
    else NA
  }),
  row.names = NULL
)

DT::datatable(RID_cpg,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation info for each read'),options = list(pageLength = 10) )
```

This results in two groups of reads: those containing mCpG and those without methylation. We then compare footprint frequencies between the two groups.

We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome.

```{r}
fps <- res$fiberHMM_fps_infire
fps_TF <- fps[fps$class %in% c("< 40bp", "40-60bp"),]
fps_gr <- GRanges(fps_TF$chr, IRanges(fps_TF$start, fps_TF$end), strand=fps_TF$strand)
mcols(fps_gr) <- fps_TF[, !(names(fps_TF) %in% c("chr","start","end","strand"))]

hw <- GRanges(region$chr, IRanges(start_motif1, end_motif1))
fps_gr <- fps_gr[width(pintersect(fps_gr, hw)) > 0]
rid_motif1_fp <- as.character(fps_gr$RID)

fps <- res$fiberHMM_fps
fps_nucl <- fps[fps$size %in% c("> 100bp", "60-100bp"),]
fps_gr_nucl <- GRanges(fps_nucl$chr, IRanges(fps_nucl$fp_blockStarts, fps_nucl$fp_blockEnds))
mcols(fps_gr_nucl) <- fps_nucl[, !(names(fps_nucl) %in% c("chr","start","end"))]
fps_gr_nucl <- fps_gr_nucl[width(pintersect(fps_gr_nucl, hw, ignore.strand = T)) > 0]
rid_fps_nucl <- as.character(fps_gr_nucl$RID)

# filter rid sets
rid_motif1_without_nucl <- setdiff(rid_motif1, rid_fps_nucl)
RIDs_overlap_both <- intersect(rid_motif1_without_nucl, RID_cpg$RID)

RIDs_access.df <- data.frame(
  RID = RIDs_overlap_both,
  TF = RIDs_overlap_both %in% rid_motif1_fp,
  mCpG = RIDs_overlap_both %in% RID_cpg$RID[RID_cpg$status=="mCpG"]
)

RIDs_access.df$TF <- factor(RIDs_access.df$TF, levels=c(FALSE, TRUE))
RIDs_access.df$mCpG <- factor(RIDs_access.df$mCpG, levels=c(FALSE, TRUE))

print(RIDs_access.df)

contingency_table <- table(RIDs_access.df[,c("TF","mCpG")]) 
prop_col <- round(prop.table(contingency_table, margin = 2) * 100,2)

print(contingency_table)
print(prop_col)

```









## chr2:27134098-27135200


The third example is region chr2:27134098-27135200, focusing on the motif chr2:27134667-27134678.

```{r message=FALSE,warning=FALSE, fig.height=10, fig.width=10}
library(data.table)
library(reshape2)
library(GenomicRanges)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(Biostrings)

source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/process_fiberseq_data.R")
source("/project/spott/kevinluo/Fiber_seq/fiberhub/code/plots.R")

test_range_str <- "chr2:27134100-27135200"
plot_name <- "NRF1_pairs_LCL_18508_chr2:27134100-27135200"
result_dir <- "/project/xinhe/xsun/fiberseq/1.TFs/plot_region/results/NRF1_pairs/NRF1_pairs_LCL_18508_chr2:27134100-27135200/parsed"

region <- get_region(result_dir)
subtitle <- sprintf("%s:%s-%s", region$chr, region$start, region$end)

# motif coordinates
start_motif1 <- 27134667
end_motif1   <- 27134678

region_plot <- list(chr ="chr10",
                    start = 27134500,
                    end = 27134800)

res <- try(assemble_region_result(result_dir,
                       filter_fire_reads = FALSE,
                       cover_region_only = FALSE,
                       select_A_only = FALSE,
                       split_haplotype = FALSE,
                       highlight_window = data.frame(start=c(start_motif1), end=c(end_motif1)),
                       color_highlight_window = "#44d6d8ff",
                       plot_name = plot_name,region = region_plot,
                       subtitle = subtitle)
  )

gg <- res$gg

print(cowplot::plot_grid(
  gg$pileup,
  gg$dimelo_reads,
  gg$fire_fdr,
  gg$fire_reads,
  gg$fiberHMM,
  ncol = 1,
  rel_heights = c(4,8,2,8,8),
  align = "v",
  axis = "b"
))


```

Next, we identify CpG positions within this motif. The CpG positions are extracted from the corresponding .fa files. 

```{r}
seq_region <- readDNAStringSet(file.path(result_dir,"/region.fa"))
dna_seq <- seq_region[[1]]
cpg_rel_pos <- start(matchPattern("CG", dna_seq))
header <- names(seq_region)       
parts <- strsplit(header, "[:-]")[[1]]
chrom <- parts[1]
region_start <- as.integer(parts[2])
cpg_df <- data.frame(
  chrom = chrom,
  position = region_start + cpg_rel_pos - 1
)
cpg_df_inmotif <- cpg_df[cpg_df$position >= start_motif1 & cpg_df$position <= end_motif1,]

rid_motif1 <- as.character(res$rids_df$RID[res$rids_df$start <= end_motif1 & res$rids_df$end >= start_motif1])

rid_motifsite_mat <- matrix(
  NA, 
  nrow = length(rid_motif1), 
  ncol = length(start_motif1:end_motif1),
  dimnames = list(rid_motif1, as.character(start_motif1:end_motif1))
)
rid_motifsite_methy <- rid_motifsite_mat

# Mark CpG positions
rid_motifsite_mat[,colnames(rid_motifsite_mat) %in% as.character(cpg_df_inmotif$position)] <- "CpG"

DT::datatable(rid_motifsite_mat,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','CpG positions in the motif for each read'),options = list(pageLength = 10) )


```


We then retrieve methylation information from res$read.

```{r}

# Fill methylation
reads_cpg <- res$reads[res$reads$base == "CG",]
reads_cpg$RID <- as.character(reads_cpg$RID)

idx <- which(reads_cpg$RID %in% rownames(rid_motifsite_methy) &
               reads_cpg$ref_position %in% colnames(rid_motifsite_methy))

for (j in idx) {
  rid_motifsite_methy[reads_cpg$RID[j], as.character(reads_cpg$ref_position[j])] <- reads_cpg$mod_qual[j]
}

DT::datatable(rid_motifsite_methy,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation quality score for each read'),options = list(pageLength = 10) )

stopifnot(identical(dimnames(rid_motifsite_methy), dimnames(rid_motifsite_mat)))

```

By combining the CpG position matrix with the methylation quality matrix, we annotate methylated sites as mCpG. A read is classified as mCpG if it contains at least one methylated CpG.

```{r}
# Combine CpG & methylation
rid_motifsite_combined <- ifelse(
  !is.na(rid_motifsite_methy) & !is.na(rid_motifsite_mat), 
  "mCpG",
  ifelse(!is.na(rid_motifsite_mat), "CpG", NA)
)

DT::datatable(rid_motifsite_combined,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation annotation for each read'),options = list(pageLength = 10) )

```

If a read has at least one mCpG, we annotate this read as mCpG

```{r}
RID_cpg <- data.frame(
  RID = rownames(rid_motifsite_combined),
  status = apply(rid_motifsite_combined, 1, function(x) {
    if ("mCpG" %in% x) "mCpG"
    else if (all(na.omit(x) == "CpG")) "CpG"
    else NA
  }),
  row.names = NULL
)

DT::datatable(RID_cpg,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Methylation info for each read'),options = list(pageLength = 10) )
```

This results in two groups of reads: those containing mCpG and those without methylation. We then compare footprint frequencies between the two groups.

We restrict the analysis to reads without nucleosome footprints, since transcription factors cannot bind when the motif is occupied by a nucleosome.

```{r}
fps <- res$fiberHMM_fps_infire
fps_TF <- fps[fps$class %in% c("< 40bp", "40-60bp"),]
fps_gr <- GRanges(fps_TF$chr, IRanges(fps_TF$start, fps_TF$end), strand=fps_TF$strand)
mcols(fps_gr) <- fps_TF[, !(names(fps_TF) %in% c("chr","start","end","strand"))]

hw <- GRanges(region$chr, IRanges(start_motif1, end_motif1))
fps_gr <- fps_gr[width(pintersect(fps_gr, hw)) > 0]
rid_motif1_fp <- as.character(fps_gr$RID)

fps <- res$fiberHMM_fps
fps_nucl <- fps[fps$size %in% c("> 100bp", "60-100bp"),]
fps_gr_nucl <- GRanges(fps_nucl$chr, IRanges(fps_nucl$fp_blockStarts, fps_nucl$fp_blockEnds))
mcols(fps_gr_nucl) <- fps_nucl[, !(names(fps_nucl) %in% c("chr","start","end"))]
fps_gr_nucl <- fps_gr_nucl[width(pintersect(fps_gr_nucl, hw, ignore.strand = T)) > 0]
rid_fps_nucl <- as.character(fps_gr_nucl$RID)

# filter rid sets
rid_motif1_without_nucl <- setdiff(rid_motif1, rid_fps_nucl)
RIDs_overlap_both <- intersect(rid_motif1_without_nucl, RID_cpg$RID)

RIDs_access.df <- data.frame(
  RID = RIDs_overlap_both,
  TF = RIDs_overlap_both %in% rid_motif1_fp,
  mCpG = RIDs_overlap_both %in% RID_cpg$RID[RID_cpg$status=="mCpG"]
)

RIDs_access.df$TF <- factor(RIDs_access.df$TF, levels=c(FALSE, TRUE))
RIDs_access.df$mCpG <- factor(RIDs_access.df$mCpG, levels=c(FALSE, TRUE))

print(RIDs_access.df)

contingency_table <- table(RIDs_access.df[,c("TF","mCpG")]) 
prop_col <- round(prop.table(contingency_table, margin = 2) * 100,2)

print(contingency_table)
print(prop_col)

```








# Summary of all motifs analyzed


```{r message=FALSE,warning=FALSE, fig.height=5, fig.width=5}

contingency_prop_col <- readRDS("/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_pairs_cpg/contingency_prop_col.RDS")

mat <- c()
names <- c()
for (i in 1:length(contingency_prop_col)){
  
  table <- contingency_prop_col[[i]]
  
  if(is.null(table) || (length(table) == 1 && is.na(table))) next
  
  extract <- contingency_prop_col[[i]][2,]
  names <- c(names,names(contingency_prop_col[i]))
  mat <- rbind(mat,extract)
  
}

mat <- as.data.frame(mat)
rownames(mat) <- NULL
colnames(mat) <- c("methylated","not_methylated")
mat$motif <- names


contingency_table <- readRDS("/project/xinhe/xsun/fiberseq/1.TFs/results/NRF1_pairs_cpg/contingency_table.RDS")

mat_table <- c()
mat_table_nfp <- c()
names <- c()
for (i in 1:length(contingency_prop_col)){
  
  table <- contingency_table[[i]]
  
  if(is.null(table) || (length(table) == 1 && is.na(table))) next
  
  extract <- contingency_table[[i]][2,]
  names <- c(names,names(contingency_table[i]))
  mat_table <- rbind(mat_table,extract)
  mat_table_nfp <- rbind(mat_table_nfp,contingency_table[[i]][1,])
  
}

mat_table <- as.data.frame(mat_table)
rownames(mat_table) <- NULL
colnames(mat_table) <- c("methylated","not_methylated")
mat_table$motif <- names

mat_table_nfp <- as.data.frame(mat_table_nfp)
rownames(mat_table_nfp) <- NULL
colnames(mat_table_nfp) <- c("methylated","not_methylated")
mat_table_nfp$motif <- names


tmp <- merge(mat_table, mat_table_nfp, by = "motif")
combine <- merge(tmp, mat, by = "motif")

colnames(combine) <- c("motifsite","#of_fp_detected_in_methy_group","#of_fp_detected_in_non-methy_group","#of_no_fp_reads_in_methy_group","#of_no_fp_reads_in_non-methy_group","%of_fp_detected_in_methy_group","%of_fp_detected_in_non-methy_group")

DT::datatable(combine,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','footprint detected, %of_fp_detected_in_methy_group is calculated as: #of_fp_detected_in_methy_group/(#of_fp_detected_in_methy_group+#of_fp_detected_in_non-methy_group)\n'),options = list(pageLength = 10) )


df_long <- mat[,1:2] %>%
  pivot_longer(cols = everything(), names_to = "CpG", values_to = "% of Footprint")

ggplot(df_long, aes(x = CpG, y = `% of Footprint`, fill = `% of Footprint`)) +
  geom_boxplot() +
  theme_minimal()


```
